---
title: "BRAF-Regulon-v2"
author: "Taras Yuziv & Lea Lemler"
date: "2025-04-09"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
# BRAF REGULON FINAL DEGREE PROJECT
# ------------------------------------------------------------------------------

Loading Libraries
```{r}
library(readr)
library(SummarizedExperiment)
library(ggplot2)
library(dplyr)
library(DESeq2)
library(maftools)
library(readxl)
library(biomaRt)
library(rtracklayer)
library(reshape2)
```

Load Data
```{r}
gene_expression_data <- read_rds("/CCBdata/data/datasets/TCGA_v38/COAD/Gene_Expression/COAD_TCGA_filtered_gene-expression.RDS")
mutation_data <- read_rds("/CCBdata/data/datasets/TCGA_v38/COAD/SNV/COAD_TCGA_filtered_snv.maf.RDS")
metadata <-readr::read_tsv("/CCBdata/users/lea/BRAF_Sex/CRC_Cohorts/Silvana/215982clinic_sup_grassso_cancerDisc18.tsv")
metadata_COAD <- read.csv("/CCBdata/users/lea/BRAF_Sex/CRC_Cohorts/Silvana/COAD_clinic.csv")
metadata_PANCAN<-readr::read_tsv("/CCBdata/users/lea/BRAF_Sex/CRC_Cohorts/Silvana/clinical_PANCAN_patient_with_followup.tsv")
MTAB_counts <- read.table(gzfile("/CCBdata/users/lea/BRAF_Postdoc_Project/Datasets/RNA-seq/E-MTAB-12862/CRC.SW.mRNA.count.txt.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)
metadata_MTAB<- read_xlsx("/CCBdata/users/lea/BRAF_Postdoc_Project/Datasets/RNA-seq/E-MTAB-12862/metadata_2.xlsx")
```

```{r}
mutation_data <- read.maf(mutation_data)
```

```{r}
expression_unstranded_data <- assays(gene_expression_data)[["unstranded"]]
```
# Preparing data TCGA
Mutation types 
```{r}
mutation_data <- mutation_data@data
V600E_BRAF_mutations <- mutation_data %>% filter(mutation_data$Hugo_Symbol == "BRAF" & mutation_data$HGVSp_Short == "p.V640E") %>% dplyr::select(Hugo_Symbol, HGVSp_Short, Tumor_Sample_Barcode)

# double check because there might be classes.
non_V600E_BRAF_mutations <- mutation_data %>% filter(mutation_data$Hugo_Symbol == "BRAF" & mutation_data$HGVSp_Short != "p.V640E") %>% dplyr::select(Hugo_Symbol, HGVSp_Short, Tumor_Sample_Barcode)
wt_BRAF_mutations <- mutation_data %>% filter(mutation_data$Hugo_Symbol != "BRAF") %>% dplyr::select(Hugo_Symbol, HGVSp_Short, Tumor_Sample_Barcode)
```

```{r}
#V600E BRAF
#I create a shorter barcode to match them with the sample names of the expression (they are unique if we take 16 characters)
V600E_BRAF_mutations$short_barcode <- substr(V600E_BRAF_mutations$Tumor_Sample_Barcode, 1, 16)
expression_colnames_short <- substr(colnames(expression_unstranded_data), 1, 16)

#And I selected the samples of the V600E BRAF
selected_columns_V600E <- expression_colnames_short %in% V600E_BRAF_mutations$short_barcode
filtered_V600E_expression <- expression_unstranded_data[, selected_columns_V600E]

#non-V600E BRAF
non_V600E_BRAF_mutations$short_barcode <- substr(non_V600E_BRAF_mutations$Tumor_Sample_Barcode, 1, 16)
#expression_colnames_short <- substr(colnames(expression_unstranded_data), 1, 16)

selected_columns_nonV600E <- expression_colnames_short %in% non_V600E_BRAF_mutations$short_barcode
filtered_nonV600E_expression <- expression_unstranded_data[, selected_columns_nonV600E]



#Wild type BRAF
wt_BRAF_mutations$short_barcode <- substr(wt_BRAF_mutations$Tumor_Sample_Barcode, 1, 16)
#expression_colnames_short <- substr(colnames(expression_unstranded_data), 1, 16)

selected_columns_wtBRAF <- expression_colnames_short %in% wt_BRAF_mutations$short_barcode
filtered_wtBRAF_expression <- expression_unstranded_data[, selected_columns_wtBRAF]
```


```{r}
duplicated_samples_V600E_nonV600E <- intersect(colnames(filtered_nonV600E_expression), colnames(filtered_V600E_expression))
duplicated_samples_V600E_Wild <- intersect(colnames(filtered_V600E_expression), colnames(filtered_wtBRAF_expression))
duplicated_samples_nonV600E_Wild <- intersect(colnames(filtered_nonV600E_expression), colnames(filtered_wtBRAF_expression))
```

Remove samples
```{r}
columns_to_remove <- duplicated_samples_V600E_nonV600E

filtered_V600E_expression <- filtered_V600E_expression[, 
  !(colnames(filtered_V600E_expression) %in% columns_to_remove)]

filtered_nonV600E_expression <- filtered_nonV600E_expression[, 
  !(colnames(filtered_nonV600E_expression) %in% columns_to_remove)]

columns_to_remove_V600E_Wildtype <- duplicated_samples_V600E_Wild

filtered_wtBRAF_expression <- filtered_wtBRAF_expression[, 
  !(colnames(filtered_wtBRAF_expression) %in% columns_to_remove_V600E_Wildtype)]

columns_to_remove_nonV600E_Wildtype <- duplicated_samples_nonV600E_Wild


filtered_wtBRAF_expression <- filtered_wtBRAF_expression[, 
  !(colnames(filtered_wtBRAF_expression) %in% columns_to_remove_nonV600E_Wildtype)]

```


## Metadata TCGA

```{r}
sample_info_wt_vs_v600e <- data.frame(
  sample = c(colnames(filtered_wtBRAF_expression),
             colnames(filtered_V600E_expression),
             colnames(filtered_nonV600E_expression)),
  condition = c(rep("wt", ncol(filtered_wtBRAF_expression)),
                rep("BRAFV600E", ncol(filtered_V600E_expression)),
                rep("BRAFnonV600E", ncol(filtered_nonV600E_expression))))
```


```{r}
sample_info_wt_vs_v600e$short_sample <- substr(sample_info_wt_vs_v600e$sample, 1, 12)
metadata_sex <- metadata_COAD %>% filter(metadata_COAD$bcr_patient_barcode %in% sample_info_wt_vs_v600e$short_sample) %>% dplyr::select(bcr_patient_barcode, gender)
metadata_msistatus <- metadata %>% filter(metadata$Sample %in% sample_info_wt_vs_v600e$short_sample) %>% dplyr::select(Sample, MsiStatus)
metadata_tumorpurity <- metadata %>% filter(metadata$Sample %in% sample_info_wt_vs_v600e$short_sample) %>% dplyr::select(Sample, TumorPurity)
metadata_age <- metadata_COAD %>% filter(metadata_COAD$bcr_patient_barcode %in% sample_info_wt_vs_v600e$short_sample) %>% dplyr::select(bcr_patient_barcode, age_at_initial_pathologic_diagnosis)
metadata_stage <- metadata_PANCAN %>% filter(metadata_PANCAN$bcr_patient_barcode %in% sample_info_wt_vs_v600e$short_sample) %>% dplyr::select(bcr_patient_barcode, pathologic_stage)
metadata_location <- metadata_PANCAN %>% filter(metadata_PANCAN$bcr_patient_barcode %in% sample_info_wt_vs_v600e$short_sample) %>% dplyr::select(bcr_patient_barcode, anatomic_neoplasm_subdivision)
# metadata_location_2 <- metadata_PANCAN %>% filter(metadata_PANCAN$bcr_patient_barcode %in% sample_info_wt_vs_v600e$short_sample) %>% dplyr::select(bcr_patient_barcode, anatomic_neoplasm_subdivision)
```

```{r}
#Left
metadata_location$Site <- ""
metadata_location$Site[metadata_location$anatomic_neoplasm_subdivision == "Sigmoid Colon"] = "left"
metadata_location$Site[metadata_location$anatomic_neoplasm_subdivision=="Descending Colon"] = "left"
metadata_location$Site[metadata_location$anatomic_neoplasm_subdivision=="Splenic Flexure"] = "left"

#Right
metadata_location$Site[metadata_location$anatomic_neoplasm_subdivision=="Cecum"] = "right"
metadata_location$Site[metadata_location$anatomic_neoplasm_subdivision=="Ascending Colon"] = "right"
metadata_location$Site[metadata_location$anatomic_neoplasm_subdivision=="Hepatic Flexure"] = "right"
metadata_location$Site[metadata_location$anatomic_neoplasm_subdivision=="Transverse Colon"] = "right"

#Stages
metadata_stage$STAGEn <- 0
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage I"] = 1
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IA"] = 1
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IB"] = 1
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage II"] = 2
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IIA"] = 2
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IIB"] = 2
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IIC"] = 2
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage III"] = 3
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IIIA"] = 3
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IIIB"] = 3
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IIIC"] = 3
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IV"] = 4
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IVA"] = 4
metadata_stage$STAGEn[metadata_stage$pathologic_stage=="Stage IVB"] = 4
```

Creation of the final metadata

```{r}
final_metadata_wt_vs_v600e <- sample_info_wt_vs_v600e %>%
  left_join(metadata_sex, by = c("short_sample" = "bcr_patient_barcode")) %>%
  left_join(metadata_msistatus, by = c("short_sample" = "Sample")) %>%
  left_join(metadata_tumorpurity, by = c("short_sample" = "Sample")) %>%
  left_join(metadata_age, by = c("short_sample" = "bcr_patient_barcode")) %>%
  left_join(metadata_stage, by = c("short_sample" = "bcr_patient_barcode")) %>%
  left_join(metadata_location, by = c("short_sample" = "bcr_patient_barcode"))

final_metadata_wt_vs_v600e <- final_metadata_wt_vs_v600e %>% dplyr::select(-short_sample, -pathologic_stage)

```

# PreMSI
```{r}
library(PreMSIm)

expression_unstranded_data_tpm <- assays(gene_expression_data)[["tpm_unstrand"]]

df_expression_unstranded_data_tpm <- data.frame(row.names = row.names(expression_unstranded_data_tpm),
                                                expression_unstranded_data_tpm)
#I take off the .X after the ensemble IDs
df_expression_unstranded_data_tpm$Gene_ID <- sub("\\..*", "", rownames(df_expression_unstranded_data_tpm))

df_cleaned <- df_expression_unstranded_data_tpm %>% group_by(Gene_ID) %>%
  summarise(across(where(is.numeric), mean, na.rm = TRUE)) %>% ungroup()

```

```{r}
#Translating the ENSMBL ID to Gene Symbols
gtf_file <- "/CCBdata/data/datasets/general_info/gencode.v47.basic.annotation.gtf"
genes_gtf <- rtracklayer::import(gtf_file)
genes <- genes_gtf[genes_gtf$type == "gene"]

gene_info <- data.frame(
  gene_id = genes$gene_id,
  gene_name = genes$gene_name
)

gene_info$gene_id <- gsub("\\..*", "", gene_info$gene_id)
df_cleaned <- merge(df_cleaned, gene_info, by.x = "Gene_ID", by.y = "gene_id", all.x = TRUE)
df_cleaned$gene_name[is.na(df_cleaned$gene_name)] <- df_cleaned$Gene_ID[is.na(df_cleaned$gene_name)]
df_cleaned$gene_name <- make.names(df_cleaned$gene_name, unique = TRUE)
row.names(df_cleaned) <- df_cleaned$gene_name
data_cleaned_MSI <- df_cleaned %>% dplyr::select(-Gene_ID, -gene_name)
```

```{r}
write.table(data_cleaned_MSI, "df_expression_unstranded_data_tpm.txt", sep = "\t")

path = system.file("extdata", "df_expression_unstranded_data_tpm.txt", package = "PreMSIm", mustWork = TRUE)
input_data = data_pre(path, type = "Symbol")
results_PreMSI <- msi_pre(input_data)
results_PreMSI$Sample <- gsub("\\.", "-", results_PreMSI$Sample)
```

1 Indicates MSI-High these samples have high levels of microsatellite
instability, which is often associated with defective DNA mismatch
repair (MMR). 
0 Indicates MSI-Low these samples have low or no
microsatellite instability, meaning their DNA mismatch repair system is
functioning normally.

```{r}
merged_data_wt_vs_v600e <- merge(results_PreMSI, final_metadata_wt_vs_v600e, 
                     by.x = "Sample", by.y = "sample", 
                     all.x = TRUE) 
merged_data_wt_vs_v600e <- merged_data_wt_vs_v600e %>% filter(Sample %in% final_metadata_wt_vs_v600e$sample)


merged_data_wt_vs_v600e <- merged_data_wt_vs_v600e %>%
  mutate(MsiStatus = ifelse(is.na(MsiStatus), 
                            ifelse(MSI_status == 1, "MSI-H", "MSS"), 
                            MsiStatus)) %>% 
  mutate(MsiStatus = ifelse(MsiStatus %in% c("POLE", "MSI-H/POLE"), 
                          ifelse(MSI_status == 1, "MSI-H", "MSS"), 
                          MsiStatus))
merged_data_wt_vs_v600e <- merged_data_wt_vs_v600e %>% dplyr::select(-MSI_status)
row.names(merged_data_wt_vs_v600e) <- merged_data_wt_vs_v600e$Sample
merged_data_wt_vs_v600e <- merged_data_wt_vs_v600e[, -1]
```

# Exploratory analysis PCA
```{r}
counts_wt_vs_V600E <- cbind(filtered_wtBRAF_expression,
                    filtered_V600E_expression,
                    filtered_nonV600E_expression)
merged_data_wt_vs_v600e <- merged_data_wt_vs_v600e[colnames(counts_wt_vs_V600E), , drop=FALSE]

```

## TCGA
```{r}
dds <- DESeqDataSetFromMatrix(countData = counts_wt_vs_V600E, 
                              colData = merged_data_wt_vs_v600e, 
                              design = ~ condition)
```

```{r}
vsd <- vst(dds, blind = FALSE)
normalized_counts <- assay(vsd)
```

```{r}
variable_genes <- apply(normalized_counts, 1, var) > 0
normalized_counts_filtered <- normalized_counts[variable_genes, ]

normalized_counts_filtered_transposed <- t(normalized_counts_filtered)
pca_result <- prcomp(normalized_counts_filtered_transposed, center = TRUE, scale. = TRUE)
```

```{r}
pca_df <- data.frame(
  Sample = rownames(normalized_counts_filtered_transposed),
  PC1 = pca_result$x[,3],
  PC2 = pca_result$x[,4],
  Site = merged_data_wt_vs_v600e$Site,
  MSI_status = merged_data_wt_vs_v600e$MsiStatus,
  Gender = merged_data_wt_vs_v600e$gender,
  TumorPurity = merged_data_wt_vs_v600e$TumorPurity,
  condition = merged_data_wt_vs_v600e$condition,
  Site_2 = merged_data_wt_vs_v600e$anatomic_neoplasm_subdivision
)
```

```{r}
ggplot(pca_df, aes(x = PC1, y = PC2, color = condition)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis of Tumor Site (VST Normalized)",
       x = "PC1", y = "PC2") +
  theme_minimal()
```


```{r}
pc1_mean <- mean(pca_df$PC1)
pc1_sd <- sd(pca_df$PC1)
pc2_mean <- mean(pca_df$PC2)
pc2_sd <- sd(pca_df$PC2)

# Define threshold for outliers 
threshold <- 2

# Filter out samples that are extreme outliers
pca_df_filtered <- pca_df[
  abs(pca_df$PC1 - pc1_mean) < threshold * pc1_sd & 
  abs(pca_df$PC2 - pc2_mean) < threshold * pc2_sd, 
]

ggplot(pca_df_filtered, aes(x = PC1, y = PC2, color = MSI_status, shape = Gender)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis TCGA",
       x = "PC1", y = "PC2") +
  theme_minimal()

ggplot(pca_df_filtered, aes(x = PC1, y = PC2, color = Site, shape = Gender)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis TCGA",
       x = "PC1", y = "PC2") +
  theme_minimal()
```

```{r}
ggplot(pca_df_filtered, aes(x = PC1, y = PC2, color = Site_2, shape = Gender)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis TCGA",
       x = "PC1", y = "PC2") +
  theme_minimal()

female_TCGA <- pca_df_filtered %>% filter(pca_df_filtered$Gender == "FEMALE")
ggplot(female_TCGA, aes(x = PC1, y = PC2, color = Site_2, shape = Gender)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis TCGA FEMALES",
       x = "PC1", y = "PC2") +
  theme_minimal()

male_TCGA <- pca_df_filtered %>% filter(pca_df_filtered$Gender == "MALE")
ggplot(male_TCGA, aes(x = PC1, y = PC2, color = Site_2, shape = Gender)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis TCGA MALES",
       x = "PC1", y = "PC2") +
  theme_minimal()
```


## MTAB
```{r}
MTAB_counts <- read.table(gzfile("/CCBdata/users/lea/BRAF_Postdoc_Project/Datasets/RNA-seq/E-MTAB-12862/CRC.SW.mRNA.count.txt.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)
metadata_MTAB<- read_xlsx("/CCBdata/users/lea/BRAF_Postdoc_Project/Datasets/RNA-seq/E-MTAB-12862/metadata_2.xlsx")
MTAB_counts_symbol <- read.table(gzfile("/CCBdata/users/lea/BRAF_Postdoc_Project/Datasets/RNA-seq/E-MTAB-12862/CRC.SW.mRNA.symbol.count.txt.gz"), sep="\t", header=TRUE, stringsAsFactors=FALSE)

```

```{r}
MTAB_counts_filtered <- MTAB_counts
colnames(MTAB_counts_filtered) <- gsub("\\.", "-", colnames(MTAB_counts_filtered))
row.names(MTAB_counts_filtered) <- MTAB_counts_filtered$`ENSG-ID`
colnames_cleaned <- gsub("CRC-SW-(U[0-9]+)-[TN]", "\\1", colnames(MTAB_counts_filtered))
colnames(MTAB_counts_filtered) <- colnames_cleaned
```

```{r}
metadata_MTAB_filtered <- metadata_MTAB[metadata_MTAB$`Sample ID` %in% colnames_cleaned, ]
row.names(metadata_MTAB_filtered) <- metadata_MTAB_filtered$`Sample ID`
```

```{r}
MTAB_counts_filtered <- MTAB_counts_filtered[, metadata_MTAB_filtered$`Sample ID`]
```

```{r}
dds_MTAB <- DESeqDataSetFromMatrix(countData = MTAB_counts_filtered, 
                                   colData = metadata_MTAB_filtered, 
                                   design = ~1)
```

```{r}
vsd_MTAB <- vst(dds_MTAB, blind = FALSE)
normalized_counts_MTAB <- assay(vsd_MTAB)
```

```{r}
variable_genes_MTAB <- apply(normalized_counts_MTAB, 1, var) > 0
normalized_counts_filtered_MTAB <- normalized_counts_MTAB[variable_genes_MTAB, ]

normalized_counts_filtered_MTAB_TRANSPOSED <- t(normalized_counts_filtered_MTAB)
pca_result_mtab <- prcomp(normalized_counts_filtered_MTAB_TRANSPOSED, center = TRUE, scale. = TRUE)

```

```{r}
pca_mtab_df <- data.frame(
  Sample = rownames(normalized_counts_filtered_MTAB_TRANSPOSED),
  PC1 = pca_result_mtab$x[,3],
  PC2 = pca_result_mtab$x[,4],
  Site = metadata_MTAB_filtered$`Anatomic Organ Subdivision`,
  MSI_status = metadata_MTAB_filtered$`MSI Status`,
  Gender = metadata_MTAB_filtered$Sex,
  CMS = metadata_MTAB_filtered$`CMS Tumour`,
  CRPS = metadata_MTAB_filtered$`CRPS Tumour`,
  site_3 = metadata_MTAB_filtered$`Tumour Site`,
  stage = metadata_MTAB_filtered$`Tumour Stage`
)
pca_mtab_df <- pca_mtab_df %>% filter(!Site == "Rectum")
```


```{r}
ggplot(mapping = aes(x = PC1, y = PC2, color = MSI_status, fill = Site)) +
  geom_point(data = pca_mtab_df[pca_mtab_df$Gender == "Female", ], 
             aes(x = PC1, y = PC2, color = MSI_status, fill = Site), 
             size = 3, alpha = 0.7, shape = 21) + 
  geom_point(data = pca_mtab_df[pca_mtab_df$Gender == "Male", ], 
             aes(x = PC1, y = PC2, color = MSI_status, fill = Site), 
             size = 3, alpha = 0.7, shape = 25) + 
  labs(title = "PCA Analysis of MTAB",
       x = "PC1", y = "PC2") +
  theme_minimal() +
  scale_fill_brewer(palette = "Set2")
 
```

```{r}
ggplot(pca_mtab_df, aes(x = PC1, y = PC2, color = MSI_status, shape = Gender)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis of MTAB",
       x = "PC1", y = "PC2") +
  theme_minimal()
```



```{r}
ggplot(pca_mtab_df, aes(x = PC1, y = PC2, color = site_3, shape = Gender)) +
  geom_point(size = 3) +
  labs(title = "PCA Analysis of MTAB",
       x = "PC3", y = "PC4") +
  theme_minimal()
```

#PAMr

```{r}
library(pamr)
library(caret)
```

```{r}
metadata_MTAB_filtered_PAMR <- metadata_MTAB_filtered %>% filter(!metadata_MTAB_filtered$`Tumour Site` == "Rectum")
normalized_counts_filtered_MTAB_PAMR <- normalized_counts_filtered_MTAB[, colnames(normalized_counts_filtered_MTAB) %in% metadata_MTAB_filtered_PAMR$`Sample ID`]
normalized_counts_filtered_MTAB_PAMR_df <- as.data.frame(normalized_counts_filtered_MTAB_PAMR)

#same genes in tcga and mtab
normalized_counts_filtered_MTAB_PAMR_df$genes <- rownames(normalized_counts_filtered_MTAB_PAMR_df)

normalized_PAM_TCGA_classifier <- sub("\\..*", "", rownames(counts_wt_vs_V600E))
common_TCGA_MTAB_genes <- intersect(normalized_counts_filtered_MTAB_PAMR_df$genes, normalized_PAM_TCGA_classifier)
normalized_counts_filtered_MTAB_PAMR_df <- normalized_counts_filtered_MTAB_PAMR_df %>% filter(normalized_counts_filtered_MTAB_PAMR_df$genes %in% common_TCGA_MTAB_genes)

normalized_counts_filtered_MTAB_PAMR_df <- normalized_counts_filtered_MTAB_PAMR_df %>% dplyr::select(-genes)
normalized_counts_filtered_MTAB_PAMR_genes_filtered <- as.matrix(normalized_counts_filtered_MTAB_PAMR_df)


X <- t(normalized_counts_filtered_MTAB_PAMR_genes_filtered)#Transpose to have samples as rows
Y <- as.factor(metadata_MTAB_filtered_PAMR$`Tumour Site`)

rownames(X) <- metadata_MTAB_filtered_PAMR$`Sample ID`
table(Y)
```

```{r}
pam_data <- list(
  x = t(X), 
  y = Y, 
  geneid = rownames(normalized_counts_filtered_MTAB_PAMR_df)
)

pam_model <- pamr.train(pam_data)
```

```{r}
pam_cv <- pamr.cv(pam_model, pam_data)
pamr.plotcv(pam_cv)
```
```{r}
optimal_threshold <- pam_cv$threshold[which.min(pam_cv$error)]
```

```{r}
important_genes <- pamr.listgenes(pam_model, pam_data, threshold = optimal_threshold)
imporatnt_genes_df <- as.data.frame(important_genes)

write.csv(imporatnt_genes_df, "important_genes_classifier.csv")
```

```{r}
save(pam_model, optimal_threshold, pam_data, file = "PAM_MTAB_model.RData")
```

```{r}
dds_PAM_TCGA_classifier <- DESeqDataSetFromMatrix(countData = counts_wt_vs_V600E, 
                                  colData = merged_data_wt_vs_v600e, 
                                  design = ~ condition)
```

```{r}
vsd_PAM_TCGA_classifier <- vst(dds_PAM_TCGA_classifier, blind = FALSE)
normalized_PAM_TCGA_classifier <- assay(vsd_PAM_TCGA_classifier)
```

```{r}
rownames(normalized_PAM_TCGA_classifier) <- gsub("\\..*", "", rownames(normalized_PAM_TCGA_classifier))
```

```{r}
common_genes <- intersect(rownames(pam_data$x), rownames(normalized_PAM_TCGA_classifier))

#subset both datasets to only include common genes
new_X_filtered <- normalized_PAM_TCGA_classifier[common_genes, , drop = FALSE]
pam_data$x <- pam_data$x[common_genes, , drop = FALSE]
```


## predict in TCGA
```{r}
new_X_filtered <- new_X_filtered[rownames(pam_data$x), , drop = FALSE]
print(identical(rownames(pam_data$x), rownames(new_X_filtered)))
```

```{r}
predictions <- pamr.predict(pam_model, new_X_filtered, threshold = optimal_threshold, type = "class")

predictions_df <- data.frame(Sample = colnames(new_X_filtered), Tumor_Site_Predicted = predictions)

```

```{r}
merged_data_wt_vs_v600e_with_classifier <- merged_data_wt_vs_v600e
merged_data_wt_vs_v600e_with_classifier$SAMPLE <- rownames(merged_data_wt_vs_v600e)
colnames(predictions_df)[1] <- "SAMPLE"
merged_data_wt_vs_v600e_with_classifier <- merged_data_wt_vs_v600e_with_classifier %>% inner_join(predictions_df, by = "SAMPLE")
```


```{r}
posterior <- pamr.predict(pam_model, new_X_filtered, threshold = optimal_threshold, type="posterior")
```

```{r}
posterior <- as.data.frame(posterior)
mean(posterior$`Right Colon`)
mean(posterior$`Left Colon`) 
```
## confusion matrix caret
```{r}
merged_data_wt_vs_v600e_with_classifier$Tumor_Site_Predicted <- factor(
  ifelse(merged_data_wt_vs_v600e_with_classifier$Tumor_Site_Predicted == "Right Colon", "right", "left")
)

# Ensure both columns are factors with the same levels
merged_data_wt_vs_v600e_with_classifier$Site <- factor(merged_data_wt_vs_v600e_with_classifier$Site)

# Now both should have identical levels
levels(merged_data_wt_vs_v600e_with_classifier$Tumor_Site_Predicted)
levels(merged_data_wt_vs_v600e_with_classifier$Site)

merged_data_wt_vs_v600e_with_classifier_filtered <- merged_data_wt_vs_v600e_with_classifier %>% filter(merged_data_wt_vs_v600e_with_classifier$Site == "right" | merged_data_wt_vs_v600e_with_classifier$Site == "left")
merged_data_wt_vs_v600e_with_classifier_filtered$Site <- factor(merged_data_wt_vs_v600e_with_classifier_filtered$Site)

merged_data_wt_vs_v600e_with_classifier_filtered$Tumor_Site_Predicted <- factor(merged_data_wt_vs_v600e_with_classifier_filtered$Tumor_Site_Predicted)

# Run confusionMatrix
caret::confusionMatrix(
  merged_data_wt_vs_v600e_with_classifier_filtered$Tumor_Site_Predicted,
  merged_data_wt_vs_v600e_with_classifier_filtered$Site
)
```






# Preparing data for Diff Exp
## Final metadata TCGA
Here form the missing values I took its predicted site, and merged it with the other ones (that were already present)
```{r}
missing_diff_exp_TCGA_metadata <- merged_data_wt_vs_v600e_with_classifier %>% filter(!(merged_data_wt_vs_v600e_with_classifier$Site == "right" | merged_data_wt_vs_v600e_with_classifier$Site == "left"))
missing_diff_exp_TCGA_metadata <- missing_diff_exp_TCGA_metadata %>% dplyr::select(-Site)
colnames(missing_diff_exp_TCGA_metadata)[9] <- "Site"

diff_exp_TCGA_metadata <- merged_data_wt_vs_v600e_with_classifier %>% dplyr::select(-Tumor_Site_Predicted)
diff_exp_TCGA_metadata <- diff_exp_TCGA_metadata %>% filter(!(diff_exp_TCGA_metadata$SAMPLE %in% missing_diff_exp_TCGA_metadata$SAMPLE))
diff_exp_TCGA_metadata <- rbind(diff_exp_TCGA_metadata, missing_diff_exp_TCGA_metadata)


diff_exp_TCGA_metadata$Site[diff_exp_TCGA_metadata$SAMPLE == "TCGA-5M-AAT5-01A-21R-A41B-07"] <- "right"
final_diff_exp_TCGA_metadata <- diff_exp_TCGA_metadata
rownames(final_diff_exp_TCGA_metadata) <- final_diff_exp_TCGA_metadata$SAMPLE

final_diff_exp_TCGA_metadata <- final_diff_exp_TCGA_metadata %>% dplyr::select(-SAMPLE)
```

## Final counts TCGA
```{r}
final_diff_exp_TCGA_counts <- counts_wt_vs_V600E
```

# Diff exp TCGA
## BRAFV600E vs WT
```{r}
diff_exp_TCGA_metadata_wt_vs_V600E <- final_diff_exp_TCGA_metadata %>% filter(final_diff_exp_TCGA_metadata$condition == "wt" | final_diff_exp_TCGA_metadata$condition == "BRAFV600E")

diff_exp_TCGA_counts_wt_vs_V600E <- final_diff_exp_TCGA_counts[, colnames(final_diff_exp_TCGA_counts) %in% rownames(diff_exp_TCGA_metadata_wt_vs_V600E)]
```

```{r}
diff_exp_TCGA_metadata_wt_vs_V600E <- diff_exp_TCGA_metadata_wt_vs_V600E[
  rownames(diff_exp_TCGA_metadata_wt_vs_V600E) %in% colnames(diff_exp_TCGA_counts_wt_vs_V600E), 
]

# Subset count matrix to include only samples present in the metadata
diff_exp_TCGA_counts_wt_vs_V600E <- diff_exp_TCGA_counts_wt_vs_V600E[, 
  colnames(diff_exp_TCGA_counts_wt_vs_V600E) %in% rownames(diff_exp_TCGA_metadata_wt_vs_V600E)]

diff_exp_TCGA_metadata_wt_vs_V600E <- diff_exp_TCGA_metadata_wt_vs_V600E[
  match(colnames(diff_exp_TCGA_counts_wt_vs_V600E), rownames(diff_exp_TCGA_metadata_wt_vs_V600E)), 
]

# Check if everything is aligned correctly
all(colnames(diff_exp_TCGA_counts_wt_vs_V600E) == rownames(diff_exp_TCGA_metadata_wt_vs_V600E))
```



```{r}
diff_exp_TCGA_metadata_wt_vs_V600E <- diff_exp_TCGA_metadata_wt_vs_V600E %>%
  filter(!is.na(gender) & !is.na(MsiStatus) & !is.na(TumorPurity) & 
         !is.na(age_at_initial_pathologic_diagnosis) & !is.na(Site))

# Subset the counts matrix to match the filtered metadata
diff_exp_TCGA_counts_wt_vs_V600E <- diff_exp_TCGA_counts_wt_vs_V600E[, 
  colnames(diff_exp_TCGA_counts_wt_vs_V600E) %in% rownames(diff_exp_TCGA_metadata_wt_vs_V600E)]

# Ensure correct order
diff_exp_TCGA_metadata_wt_vs_V600E <- diff_exp_TCGA_metadata_wt_vs_V600E[
  match(colnames(diff_exp_TCGA_counts_wt_vs_V600E), rownames(diff_exp_TCGA_metadata_wt_vs_V600E)), ]


```


```{r}
dds_wt_vs_v600e <- DESeqDataSetFromMatrix(
  countData = diff_exp_TCGA_counts_wt_vs_V600E,
  colData = diff_exp_TCGA_metadata_wt_vs_V600E,
  design = ~ gender + MsiStatus + TumorPurity + age_at_initial_pathologic_diagnosis + Site + condition
)
#remove stage

# keep <- rowSums(counts(dds_wt_vs_v600e) >= 10) >= 45
# dds_wt_vs_v600e <- dds_wt_vs_v600e[keep,]
# 
# dds_wt_vs_v600e <- DESeq(dds_wt_vs_v600e)
levels(dds_wt_vs_v600e$condition)
```

```{r}
load("~/OnDemand/OnDemand_Results_TCGA/DESeq2_wt_vs_v600e_v2_results.RData")

dds_wt_vs_v600e
res_wt_vs_v600e <- results(dds_wt_vs_v600e, alpha = 0.05, contrast = c('condition', 'BRAFV600E','wt'))
res_wt_vs_v600e <- res_wt_vs_v600e[order(res_wt_vs_v600e$padj), ]
res_wt_vs_v600e_df <- as.data.frame(res_wt_vs_v600e)
res_wt_vs_v600e_df$Gene <- rownames(res_wt_vs_v600e_df)

res_wt_vs_v600e_df <- res_wt_vs_v600e_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_wt_vs_v600e_df, "DESeq2_wt_vs_v600e_v2_results.csv", row.names = FALSE)

```

```{r}
DEG_wt_vs_V600E_v2 <- read.csv("DESeq2_wt_vs_v600e_v2_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_wt_vs_V600E_v2 <- DEG_wt_vs_V600E_v2 %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_wt_vs_V600E_v2 <- DEG_wt_vs_V600E_v2 %>% filter(!is.na(log2FoldChange))
DEG_wt_vs_V600E_v2 <- DEG_wt_vs_V600E_v2 %>% filter(!is.na(gene_symbol))
# DEG_wt_vs_V600E <- DEG_wt_vs_V600E %>% filter(DEG_wt_vs_V600E$pvalue < 0.05)#get rid of the
ranked_genes_wt_vs_v600e_v2 <- DEG_wt_vs_V600E_v2$log2FoldChange
names(ranked_genes_wt_vs_v600e_v2) <- DEG_wt_vs_V600E_v2$gene_symbol

ranked_genes_wt_vs_v600e_v2 <- sort(ranked_genes_wt_vs_v600e_v2, decreasing=TRUE)
ranked_genes_wt_vs_v600e_v2 <- ranked_genes_wt_vs_v600e_v2[!duplicated(names(ranked_genes_wt_vs_v600e_v2))]
```

###VOLCANO
```{r}
library(EnhancedVolcano)

v600e_vs_wt <- EnhancedVolcano(DEG_wt_vs_V600E_v2,
                 lab = DEG_wt_vs_V600E_v2$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "BRAF V600E vs wt",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
v600e_vs_wt
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_wt_vs_v600e_v2 <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_wt_vs_v600e_v2) #not specify permutations
```

```{r}
fgsea_results_wt_vs_v600e_v2 <- fgsea_results_wt_vs_v600e_v2[order(fgsea_results_wt_vs_v600e_v2$padj), ]

# head(fgsea_results_wt_vs_v600e_v2 %>% arrange(padj), n=15)
```

### HALLMARKS
```{r}
plot_res <- fgsea_results_wt_vs_v600e_v2 %>% filter(pval < 0.05)
hallmarks_results <- ggplot(plot_res, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFV600E vs WT", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
hallmarks_results
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_wt_vs_v600e_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_wt_vs_v600e_v2)
```


```{r}
fgsea_results_wt_vs_v600e_kegg <- fgsea_results_wt_vs_v600e_kegg[order(fgsea_results_wt_vs_v600e_kegg$padj), ]
```

### KEGG
```{r}
plot_res_kegg <- fgsea_results_wt_vs_v600e_kegg %>% filter(padj < 0.1)
tcga_kegg_v600e_vs_wt <- ggplot(plot_res_kegg, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFV600E vs WT KEGG padj<0.1", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```

```{r}
msigdb_reactome <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:REACTOME")
reactome_pathway_list <- split(msigdb_reactome$gene_symbol, msigdb_reactome$gs_name)
```


```{r}
fgsea_results_wt_vs_v600e_reactome <- fgsea(pathways = reactome_pathway_list, 
                                            stats = ranked_genes_wt_vs_v600e_v2)
```


```{r}
fgsea_results_wt_vs_v600e_reactome <- fgsea_results_wt_vs_v600e_reactome[order(fgsea_results_wt_vs_v600e_reactome$padj), ]
```
### REACTOME
```{r}
fgsea_results_wt_vs_v600e_reactome_PLOT <- fgsea_results_wt_vs_v600e_reactome %>% filter(padj < 0.0005)
tcga_reactome_v600e_vs_wt<- ggplot(fgsea_results_wt_vs_v600e_reactome_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFV600E VS WT REACTOME padj<0.05", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```


##nonV600E vs WT

```{r}
load("~/OnDemand/OnDemand_Results_TCGA/DESeq2_wt_vs_nonv600e_v2_results.RData")
dds_wt_vs_nonv600e

res_wt_vs_nonv600e <- results(dds_wt_vs_nonv600e, alpha = 0.05, contrast = c('condition', 'BRAFnonV600E','wt'))
res_wt_vs_nonv600e <- res_wt_vs_nonv600e[order(res_wt_vs_nonv600e$padj), ]
res_wt_vs_nonv600e_df <- as.data.frame(res_wt_vs_nonv600e)
res_wt_vs_nonv600e_df$Gene <- rownames(res_wt_vs_nonv600e_df)

res_wt_vs_nonv600e_df <- res_wt_vs_nonv600e_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_wt_vs_nonv600e_df, "DESeq2_wt_vs_nonv600e_v2_results.csv", row.names = FALSE)
```


```{r}
DEG_wt_vs_nonV600E_v2 <- read.csv("DESeq2_wt_vs_nonv600e_v2_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_wt_vs_nonV600E_v2 <- DEG_wt_vs_nonV600E_v2 %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_wt_vs_nonV600E_v2 <- DEG_wt_vs_nonV600E_v2 %>% filter(!is.na(log2FoldChange))
DEG_wt_vs_nonV600E_v2 <- DEG_wt_vs_nonV600E_v2 %>% filter(!is.na(gene_symbol))
# DEG_wt_vs_V600E <- DEG_wt_vs_V600E %>% filter(DEG_wt_vs_V600E$pvalue < 0.05)#get rid of the
ranked_genes_wt_vs_nonv600e_v2 <- DEG_wt_vs_nonV600E_v2$log2FoldChange
names(ranked_genes_wt_vs_nonv600e_v2) <- DEG_wt_vs_nonV600E_v2$gene_symbol

ranked_genes_wt_vs_nonv600e_v2 <- sort(ranked_genes_wt_vs_nonv600e_v2, decreasing=TRUE)
ranked_genes_wt_vs_nonv600e_v2 <- ranked_genes_wt_vs_nonv600e_v2[!duplicated(names(ranked_genes_wt_vs_nonv600e_v2))]
```

###VOLCANO
```{r}
nonv600e_vs_wt <- EnhancedVolcano(DEG_wt_vs_nonV600E_v2,
                 lab = DEG_wt_vs_nonV600E_v2$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "BRAFnonV600E vs WT",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
nonv600e_vs_wt
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_wt_vs_nonv600e_v2 <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_wt_vs_nonv600e_v2) #not specify permutations
```

```{r}
fgsea_results_wt_vs_nonv600e_v2 <- fgsea_results_wt_vs_nonv600e_v2[order(fgsea_results_wt_vs_nonv600e_v2$padj), ]

head(fgsea_results_wt_vs_nonv600e_v2)
```

###HALLMARKS
```{r}
fgsea_results_wt_vs_nonv600e_v2_plot <- fgsea_results_wt_vs_nonv600e_v2 %>% filter(pval < 0.05)
hallmars_nonv600e <- ggplot(fgsea_results_wt_vs_nonv600e_v2_plot, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFnonV600E vs WT", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
hallmars_nonv600e
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_wt_vs_nonv600e_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_wt_vs_nonv600e_v2)
```


```{r}
fgsea_results_wt_vs_nonv600e_kegg <- fgsea_results_wt_vs_nonv600e_kegg[order(fgsea_results_wt_vs_nonv600e_kegg$padj), ]
```

### KEGG
```{r}
fgsea_results_wt_vs_nonv600e_kegg_PLOT <- fgsea_results_wt_vs_nonv600e_kegg %>% filter(pval < 0.05)
tcga_kegg_nonv600e_vvs_wt <- ggplot(fgsea_results_wt_vs_nonv600e_kegg_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFnonV600E vs WT KEGG", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```

```{r}
msigdb_reactome <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:REACTOME")
reactome_pathway_list <- split(msigdb_reactome$gene_symbol, msigdb_reactome$gs_name)
```


```{r}
fgsea_results_wt_vs_nonv600e_reactome <- fgsea(pathways = reactome_pathway_list, 
                                            stats = ranked_genes_wt_vs_nonv600e_v2)
```


```{r}
fgsea_results_wt_vs_nonv600e_reactome <- fgsea_results_wt_vs_nonv600e_reactome[order(fgsea_results_wt_vs_nonv600e_reactome$padj), ]
```
### REACTOME
```{r}
fgsea_results_wt_vs_nonv600e_reactome_PLOT <- fgsea_results_wt_vs_nonv600e_reactome %>% filter(padj < 0.05)
ggplot(fgsea_results_wt_vs_nonv600e_reactome_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFnonV600E vs WT REACTOME padj<0.05", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```

##V600E vs nonV600E
```{r}

load("~/OnDemand/OnDemand_Results_TCGA/DESeq2_v600e_vs_nonv600e_v2_results.RData")
dds_wt_v600e_nonv600e

res_v600e_vs_nonv600e <- results(dds_wt_v600e_nonv600e, alpha = 0.05, contrast = c('condition', 'BRAFV600E','BRAFnonV600E'))
res_v600e_vs_nonv600e <- res_v600e_vs_nonv600e[order(res_v600e_vs_nonv600e$padj), ]
res_v600e_vs_nonv600e_df <- as.data.frame(res_v600e_vs_nonv600e)
res_v600e_vs_nonv600e_df$Gene <- rownames(res_v600e_vs_nonv600e_df)

res_v600e_vs_nonv600e_df <- res_v600e_vs_nonv600e_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_v600e_vs_nonv600e_df, "DESeq2_v600e_vs_nonv600e_v2_results.csv", row.names = FALSE)
```

```{r}
DEG_V600E_vs_nonV600E_v2 <- read.csv("DESeq2_v600e_vs_nonv600e_v2_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_V600E_vs_nonV600E_v2 <- DEG_V600E_vs_nonV600E_v2 %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_V600E_vs_nonV600E_v2 <- DEG_V600E_vs_nonV600E_v2 %>% filter(!is.na(log2FoldChange))
DEG_V600E_vs_nonV600E_v2 <- DEG_V600E_vs_nonV600E_v2 %>% filter(!is.na(gene_symbol))
# DEG_wt_vs_V600E <- DEG_wt_vs_V600E %>% filter(DEG_wt_vs_V600E$pvalue < 0.05)#get rid of the
ranked_genes_v600e_vs_nonv600e_v2 <- DEG_V600E_vs_nonV600E_v2$log2FoldChange
names(ranked_genes_v600e_vs_nonv600e_v2) <- DEG_V600E_vs_nonV600E_v2$gene_symbol

ranked_genes_v600e_vs_nonv600e_v2 <- sort(ranked_genes_v600e_vs_nonv600e_v2, decreasing=TRUE)
ranked_genes_v600e_vs_nonv600e_v2 <- ranked_genes_v600e_vs_nonv600e_v2[!duplicated(names(ranked_genes_v600e_vs_nonv600e_v2))]
```

### VOLCANO
```{r}
v600e_nonv600e <- EnhancedVolcano(DEG_V600E_vs_nonV600E_v2,
                 lab = DEG_V600E_vs_nonV600E_v2$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "BRAF V600E vs BRAF nonV600E",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
v600e_nonv600e
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_v600e_vs_nonv600e_v2 <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_v600e_vs_nonv600e_v2) #not specify permutations
```

```{r}
fgsea_results_v600e_vs_nonv600e_v2 <- fgsea_results_v600e_vs_nonv600e_v2[order(fgsea_results_v600e_vs_nonv600e_v2$padj), ]

head(fgsea_results_v600e_vs_nonv600e_v2)
```

###HALLMARKS
```{r}
fgsea_results_v600e_vs_nonv600e_v2_plot <- fgsea_results_v600e_vs_nonv600e_v2 %>% filter(pval < 0.05)
hallmarks_combo <- ggplot(fgsea_results_v600e_vs_nonv600e_v2_plot, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFV600E vs BRAFnonV600E", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()

hallmarks_combo
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_v600e_vs_nonv600e_v2_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_v600e_vs_nonv600e_v2)
```


```{r}
fgsea_results_v600e_vs_nonv600e_v2_kegg <- fgsea_results_v600e_vs_nonv600e_v2_kegg[order(fgsea_results_v600e_vs_nonv600e_v2_kegg$padj), ]
```

### KEGG
```{r}
fgsea_results_v600e_vs_nonv600e_v2_kegg_plot <- fgsea_results_v600e_vs_nonv600e_v2_kegg %>% filter(padj < 0.1)
tcga_kegg_v600e_nonv600e <- ggplot(fgsea_results_v600e_vs_nonv600e_v2_kegg_plot, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFV600E VS BRAFnonV600E", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()


```

```{r}
msigdb_reactome <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:REACTOME")
reactome_pathway_list <- split(msigdb_reactome$gene_symbol, msigdb_reactome$gs_name)
```


```{r}
fgsea_results_v600e_vs_nonv600e_reactome <- fgsea(pathways = reactome_pathway_list, 
                                            stats = ranked_genes_v600e_vs_nonv600e_v2)
```


```{r}
fgsea_results_v600e_vs_nonv600e_reactome <- fgsea_results_v600e_vs_nonv600e_reactome[order(fgsea_results_v600e_vs_nonv600e_reactome$padj), ]
```
### REACTOME
```{r}
fgsea_results_v600e_vs_nonv600e_reactome_PLOT <- fgsea_results_v600e_vs_nonv600e_reactome %>% filter(padj < 0.00005)
ggplot(fgsea_results_v600e_vs_nonv600e_reactome_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="BRAFV600E vs BRAFnonV600E REACTOME", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```

#MTAB MAF FILE
```{r}
library(stringr)
sample_ids_MTAB <- metadata_MTAB$`DNA Tumor Sample Barcode`
grep_lines <- readLines("/CCBdata/users/lea/BRAF_Postdoc_Project/Datasets/RNA-seq/E-MTAB-12862/Mutation/results_CRC_Braf/annotation/project/braf_grep_results.txt")

```

```{r}
parse_braf_line <- function(line) {
  sample <- str_match(line, "^([^/]+)/")[,2]
  protein_change <- str_match(line, "p\\.\\w+\\d+\\w+|p\\.\\w+\\*|p\\.\\w+fs")[,1]
  data.frame(sample = sample, mutation = protein_change, stringsAsFactors = FALSE)
}
```

```{r}
parsed <- do.call(rbind, lapply(grep_lines, parse_braf_line))
```

```{r}
parsed_unique <- parsed[!duplicated(parsed$sample), ]
```

```{r}
final_df <- data.frame(sample = sample_ids_MTAB, stringsAsFactors = FALSE)
final_df <- merge(final_df, parsed_unique, by.x = "sample", by.y = "sample", all.x = TRUE)
final_df$mutation[is.na(final_df$mutation)] <- "wt"
```

```{r}
colnames(metadata_MTAB)[3] <- "DNA_Tumor_Sample_Barcode"
colnames(final_df)[1] <- "DNA_Tumor_Sample_Barcode"
metadata_MTAB_mutation <- metadata_MTAB %>%
  inner_join(final_df, by = "DNA_Tumor_Sample_Barcode")
```

## TCGA vs MTAB
```{r}
table(metadata_MTAB_mutation$mutation)
#39 nonV600E
#208 V600E
#816 wt
table(merged_data_wt_vs_v600e$condition)
```


#DIFF EXP MTAB
```{r}
metadata_MTAB_mutation_filtered <- metadata_MTAB_mutation %>% filter(!metadata_MTAB_mutation$`Anatomic Organ Subdivision` == "Rectum")
metadata_MTAB_mutation_filtered <- metadata_MTAB_mutation_filtered[, -1]

MTAB_V600E <- metadata_MTAB_mutation_filtered %>% filter(metadata_MTAB_mutation_filtered$mutation == "p.Val640Glu")

MTAB_nonV600E <- metadata_MTAB_mutation_filtered %>% filter(!(metadata_MTAB_mutation_filtered$mutation == "p.Val640Glu") & !(metadata_MTAB_mutation_filtered$mutation == "wt"))

MTAB_BRAF_WT <- metadata_MTAB_mutation_filtered %>% filter(metadata_MTAB_mutation_filtered$mutation == "wt")
#V600E 196 samples
#nonV600E 28 samples
#wt 558 samples
```

## V600E vs WT
```{r}
MTAB_V600E_metadata_vs_wt <- metadata_MTAB_mutation_filtered %>% filter((metadata_MTAB_mutation_filtered$mutation == "p.Val640Glu") | (metadata_MTAB_mutation_filtered$mutation == "wt"))

MTAB_counts_diffexp <- MTAB_counts
colnames(MTAB_counts_diffexp) <- gsub("\\.", "-", colnames(MTAB_counts_diffexp))
rownames(MTAB_counts_diffexp) <- MTAB_counts_diffexp$`ENSG-ID`
MTAB_counts_diffexp <- MTAB_counts_diffexp[, -1]

MTAB_counts_filtered_v600e_vs_wt <- MTAB_counts_diffexp[, colnames(MTAB_counts_diffexp) %in% MTAB_V600E_metadata_vs_wt$DNA_Tumor_Sample_Barcode]


```

```{r}
colnames(MTAB_V600E_metadata_vs_wt)[40] <- "MSI_status"
colnames(MTAB_V600E_metadata_vs_wt)[33] <- "Tumour_Cell_Content_Pathology"
colnames(MTAB_V600E_metadata_vs_wt)[17] <- "Age_at_diagnosis"
colnames(MTAB_V600E_metadata_vs_wt)[21] <- "Primary_Site_Disease"
colnames(MTAB_V600E_metadata_vs_wt)[23] <- "Tumour_Site"

rownames(MTAB_V600E_metadata_vs_wt) <- MTAB_V600E_metadata_vs_wt$DNA_Tumor_Sample_Barcode

remove_col_mtab <- setdiff(rownames(MTAB_V600E_metadata_vs_wt), colnames(MTAB_counts_filtered_v600e_vs_wt))
MTAB_V600E_metadata_vs_wt <- MTAB_V600E_metadata_vs_wt %>% filter(!(MTAB_V600E_metadata_vs_wt$DNA_Tumor_Sample_Barcode %in% remove_col_mtab))


MTAB_V600E_metadata_vs_wt$mutation <- as.factor(MTAB_V600E_metadata_vs_wt$mutation)
MTAB_V600E_metadata_vs_wt$MSI_status <- as.factor(MTAB_V600E_metadata_vs_wt$MSI_status)
MTAB_V600E_metadata_vs_wt$Sex <- as.factor(MTAB_V600E_metadata_vs_wt$Sex)
MTAB_V600E_metadata_vs_wt$Tumour_Cell_Content_Pathology <- as.factor(MTAB_V600E_metadata_vs_wt$Tumour_Cell_Content_Pathology)
MTAB_V600E_metadata_vs_wt$Age_at_diagnosis <- as.factor(MTAB_V600E_metadata_vs_wt$Age_at_diagnosis)
MTAB_V600E_metadata_vs_wt$Primary_Site_Disease <- as.factor(MTAB_V600E_metadata_vs_wt$Primary_Site_Disease)

rownames(MTAB_V600E_metadata_vs_wt) <- MTAB_V600E_metadata_vs_wt$DNA_Tumor_Sample_Barcode

dds_MTAB_diffexp <- DESeqDataSetFromMatrix(countData = MTAB_counts_filtered_v600e_vs_wt,
                                   colData = MTAB_V600E_metadata_vs_wt,
                                   design = ~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation)

# save(dds_MTAB_diffexp, MTAB_counts_filtered_v600e_vs_wt, MTAB_V600E_metadata_vs_wt, file = "DESeq_MTAB_V600E_vs_WT.Rdata")

# load("DESeq_MTAB_V600E_vs_WT.Rdata")
# 
# keep <- rowSums(counts(dds_MTAB_diffexp) >= 10) >= 195
# dds_MTAB_diffexp <- dds_MTAB_diffexp[keep,]
#  
# dds_MTAB_diffexp <- DESeq(dds_MTAB_diffexp)
# save(dds_MTAB_diffexp, file = "DESeq_MTAB_V600E_vs_WT.Rdata")
```

```{r}
load("DESeq_MTAB_V600E_vs_WT_results.Rdata")

dds_MTAB_diffexp
res_dds_MTAB_diffexp <- results(dds_MTAB_diffexp, alpha = 0.05, contrast = c('mutation', 'p.Val640Glu','wt'))
res_dds_MTAB_diffexp <- res_dds_MTAB_diffexp[order(res_dds_MTAB_diffexp$padj), ]
res_dds_MTAB_diffexp_df <- as.data.frame(res_dds_MTAB_diffexp)
res_dds_MTAB_diffexp_df$Gene <- rownames(res_dds_MTAB_diffexp_df)

res_dds_MTAB_diffexp_df <- res_dds_MTAB_diffexp_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_dds_MTAB_diffexp_df, "DESeq2_MTAB_V600E_vs_WT_results.csv", row.names = FALSE)
```

```{r}
DEG_MTAB_V600E_VS_WT <- read.csv("DESeq2_MTAB_V600E_vs_WT_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_MTAB_V600E_VS_WT <- DEG_MTAB_V600E_VS_WT %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_MTAB_V600E_VS_WT <- DEG_MTAB_V600E_VS_WT %>% filter(!is.na(log2FoldChange))
DEG_MTAB_V600E_VS_WT <- DEG_MTAB_V600E_VS_WT %>% filter(!is.na(gene_symbol))


ranked_genes_mtab_v600e_vs_wt <- DEG_MTAB_V600E_VS_WT$log2FoldChange
names(ranked_genes_mtab_v600e_vs_wt) <- DEG_MTAB_V600E_VS_WT$gene_symbol

ranked_genes_mtab_v600e_vs_wt <- sort(ranked_genes_mtab_v600e_vs_wt, decreasing=TRUE)
ranked_genes_mtab_v600e_vs_wt <- ranked_genes_mtab_v600e_vs_wt[!duplicated(names(ranked_genes_mtab_v600e_vs_wt))]
```

### VOLCANO
```{r}
library(EnhancedVolcano)

MTAB_v600e_vs_wt_volcano <- EnhancedVolcano(DEG_MTAB_V600E_VS_WT,
                 lab = DEG_MTAB_V600E_VS_WT$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "MTAB BRAF V600E vs wt",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
MTAB_v600e_vs_wt_volcano
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_mtab_v600e_vs_wt <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_mtab_v600e_vs_wt) #not specify permutations
```

```{r}
fgsea_results_mtab_v600e_vs_wt <- fgsea_results_mtab_v600e_vs_wt[order(fgsea_results_mtab_v600e_vs_wt$padj), ]

head(fgsea_results_mtab_v600e_vs_wt %>% arrange(padj), n=15)
```
### HALLMARKS
```{r}
plot_mtab_V600e_wt <- fgsea_results_mtab_v600e_vs_wt %>% filter(pval < 0.05)
hallmarks_results_mtab_v600e_wt <- ggplot(plot_mtab_V600e_wt, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAFV600E vs WT", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()

hallmarks_results_mtab_v600e_wt
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_MTAB_v600e_vs_wt_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_mtab_v600e_vs_wt)
```


```{r}
fgsea_results_MTAB_v600e_vs_wt_kegg <- fgsea_results_MTAB_v600e_vs_wt_kegg[order(fgsea_results_MTAB_v600e_vs_wt_kegg$padj), ]
```

### KEGG
```{r}
fgsea_results_MTAB_v600e_vs_wt_kegg_PLOT <- fgsea_results_MTAB_v600e_vs_wt_kegg %>% filter(pval < 0.05)
ggplot(fgsea_results_MTAB_v600e_vs_wt_kegg_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAFV600E vs WT KEGG", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```

```{r}
msigdb_reactome <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:REACTOME")
reactome_pathway_list <- split(msigdb_reactome$gene_symbol, msigdb_reactome$gs_name)
```


```{r}
fgsea_results_mtab_v600e_vs_wt_reactome <- fgsea(pathways = reactome_pathway_list, 
                                            stats = ranked_genes_mtab_v600e_vs_wt)
```


```{r}
fgsea_results_mtab_v600e_vs_wt_reactome <- fgsea_results_mtab_v600e_vs_wt_reactome[order(fgsea_results_mtab_v600e_vs_wt_reactome$padj), ]
```
### REACTOME
```{r}
fgsea_results_mtab_v600e_vs_wt_reactome_PLOT <- fgsea_results_mtab_v600e_vs_wt_reactome %>% filter(padj < 0.05)
ggplot(fgsea_results_mtab_v600e_vs_wt_reactome_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title=" MTAB BRAFV600E VS WT REACTOME", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```

## nonV600E vs WT
```{r}
MTAB_metadata_nonv600e_vs_wt <- metadata_MTAB_mutation_filtered %>% filter((metadata_MTAB_mutation_filtered$mutation == "wt") | !(metadata_MTAB_mutation_filtered$mutation == "p.Val640Glu"))
MTAB_metadata_nonv600e_vs_wt$mutation[MTAB_metadata_nonv600e_vs_wt$mutation != "wt"] <- "nonV600E"

MTAB_counts_diffexp <- MTAB_counts
colnames(MTAB_counts_diffexp) <- gsub("\\.", "-", colnames(MTAB_counts_diffexp))
rownames(MTAB_counts_diffexp) <- MTAB_counts_diffexp$`ENSG-ID`
MTAB_counts_diffexp <- MTAB_counts_diffexp[, -1]

MTAB_counts_filtered_nonv600e_vs_wt <- MTAB_counts_diffexp[, colnames(MTAB_counts_diffexp) %in% MTAB_metadata_nonv600e_vs_wt$DNA_Tumor_Sample_Barcode]
```

```{r}
colnames(MTAB_metadata_nonv600e_vs_wt)[40] <- "MSI_status"
colnames(MTAB_metadata_nonv600e_vs_wt)[33] <- "Tumour_Cell_Content_Pathology"
colnames(MTAB_metadata_nonv600e_vs_wt)[17] <- "Age_at_diagnosis"
colnames(MTAB_metadata_nonv600e_vs_wt)[21] <- "Primary_Site_Disease"
colnames(MTAB_metadata_nonv600e_vs_wt)[23] <- "Tumour_Site"

rownames(MTAB_metadata_nonv600e_vs_wt) <- MTAB_metadata_nonv600e_vs_wt$DNA_Tumor_Sample_Barcode
remove_col_mtab_nonv600e_vs_wt <- setdiff(rownames(MTAB_metadata_nonv600e_vs_wt), colnames(MTAB_counts_filtered_nonv600e_vs_wt))

MTAB_metadata_nonv600e_vs_wt <- MTAB_metadata_nonv600e_vs_wt %>% filter(!(MTAB_metadata_nonv600e_vs_wt$DNA_Tumor_Sample_Barcode %in% remove_col_mtab_nonv600e_vs_wt))

dds_MTAB_diffexp_nonv600e_vs_wt <- DESeqDataSetFromMatrix(countData = MTAB_counts_filtered_nonv600e_vs_wt,
                                   colData = MTAB_metadata_nonv600e_vs_wt,
                                   design = ~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation)

# save(dds_MTAB_diffexp_nonv600e_vs_wt, MTAB_counts_filtered_nonv600e_vs_wt, MTAB_metadata_nonv600e_vs_wt, file = "DESeq2_MTAB_nonV600E_vs_wt.RData")

# load("DESeq2_MTAB_nonV600E_vs_wt.RData")
# 
# keep <- rowSums(counts(dds_MTAB_diffexp_nonv600e_vs_wt) >= 10) >= 28
# dds_MTAB_diffexp_nonv600e_vs_wt <- dds_MTAB_diffexp_nonv600e_vs_wt[keep,]
#  
# dds_MTAB_diffexp_nonv600e_vs_wt <- DESeq(dds_MTAB_diffexp_nonv600e_vs_wt)
# save(dds_MTAB_diffexp_nonv600e_vs_wt, file = "DESeq2_MTAB_nonV600E_vs_wt_results.Rdata")

```

```{r}
load("DESeq2_MTAB_nonV600E_vs_wt_results.Rdata")

dds_MTAB_diffexp_nonv600e_vs_wt
res_dds_MTAB_diffexp_nonv600e_vs_wt <- results(dds_MTAB_diffexp_nonv600e_vs_wt, alpha = 0.05, contrast = c('mutation', 'nonV600E','wt'))
res_dds_MTAB_diffexp_nonv600e_vs_wt <- res_dds_MTAB_diffexp_nonv600e_vs_wt[order(res_dds_MTAB_diffexp_nonv600e_vs_wt$padj), ]
res_dds_MTAB_diffexp_nonv600e_vs_wt_df <- as.data.frame(res_dds_MTAB_diffexp_nonv600e_vs_wt)
res_dds_MTAB_diffexp_nonv600e_vs_wt_df$Gene <- rownames(res_dds_MTAB_diffexp_nonv600e_vs_wt_df)

res_dds_MTAB_diffexp_nonv600e_vs_wt_df <- res_dds_MTAB_diffexp_nonv600e_vs_wt_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_dds_MTAB_diffexp_nonv600e_vs_wt_df, "DESeq2_MTAB_nonV600E_vs_WT_results.csv", row.names = FALSE)
```

```{r}
DEG_MTAB_nonV600E_VS_WT <- read.csv("DESeq2_MTAB_nonV600E_vs_WT_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_MTAB_nonV600E_VS_WT <- DEG_MTAB_nonV600E_VS_WT %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_MTAB_nonV600E_VS_WT <- DEG_MTAB_nonV600E_VS_WT %>% filter(!is.na(log2FoldChange))
DEG_MTAB_nonV600E_VS_WT <- DEG_MTAB_nonV600E_VS_WT %>% filter(!is.na(gene_symbol))


ranked_genes_mtab_nonv600e_vs_wt <- DEG_MTAB_nonV600E_VS_WT$log2FoldChange
names(ranked_genes_mtab_nonv600e_vs_wt) <- DEG_MTAB_nonV600E_VS_WT$gene_symbol

ranked_genes_mtab_nonv600e_vs_wt <- sort(ranked_genes_mtab_nonv600e_vs_wt, decreasing=TRUE)
ranked_genes_mtab_nonv600e_vs_wt <- ranked_genes_mtab_nonv600e_vs_wt[!duplicated(names(ranked_genes_mtab_nonv600e_vs_wt))]
```

### VOLCANO
```{r}
library(EnhancedVolcano)

MTAB_nonv600e_vs_wt_volcano <- EnhancedVolcano(DEG_MTAB_nonV600E_VS_WT,
                 lab = DEG_MTAB_nonV600E_VS_WT$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "MTAB BRAF nonV600E vs wt",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
MTAB_nonv600e_vs_wt_volcano
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_mtab_nonv600e_vs_wt <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_mtab_nonv600e_vs_wt) #not specify permutations
```

```{r}
fgsea_results_mtab_nonv600e_vs_wt <- fgsea_results_mtab_nonv600e_vs_wt[order(fgsea_results_mtab_nonv600e_vs_wt$padj), ]

head(fgsea_results_mtab_nonv600e_vs_wt %>% arrange(padj), n=15)
```
### HALLMARKS
```{r}
plot_mtab_nonV600e_wt <- fgsea_results_mtab_nonv600e_vs_wt %>% filter(pval < 0.05)
hallmarks_results_mtab_nonv600e_wt <- ggplot(plot_mtab_nonV600e_wt, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAF nonV600E vs WT", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()

hallmarks_results_mtab_nonv600e_wt
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_MTAB_nonv600e_vs_wt_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_mtab_nonv600e_vs_wt)
```


```{r}
fgsea_results_MTAB_nonv600e_vs_wt_kegg <- fgsea_results_MTAB_nonv600e_vs_wt_kegg[order(fgsea_results_MTAB_nonv600e_vs_wt_kegg$padj), ]
```

### KEGG
```{r}
fgsea_results_MTAB_nonv600e_vs_wt_kegg_PLOT <- fgsea_results_MTAB_nonv600e_vs_wt_kegg %>% filter(pval < 0.05)
ggplot(fgsea_results_MTAB_nonv600e_vs_wt_kegg_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAFnonV600E vs WT KEGG", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```


```{r}
msigdb_reactome <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:REACTOME")
reactome_pathway_list <- split(msigdb_reactome$gene_symbol, msigdb_reactome$gs_name)
```


```{r}
fgsea_results_mtab_nonv600e_vs_wt_reactome <- fgsea(pathways = reactome_pathway_list, 
                                            stats = ranked_genes_mtab_nonv600e_vs_wt)
```


```{r}
fgsea_results_mtab_nonv600e_vs_wt_reactome <- fgsea_results_mtab_nonv600e_vs_wt_reactome[order(fgsea_results_mtab_nonv600e_vs_wt_reactome$padj), ]
```
### REACTOME
```{r}
fgsea_results_mtab_nonv600e_vs_wt_reactome_PLOT <- fgsea_results_mtab_nonv600e_vs_wt_reactome %>% filter(padj < 0.05)
ggplot(fgsea_results_mtab_nonv600e_vs_wt_reactome_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title=" MTAB BRAF nonV600E VS WT REACTOME", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```


## V600E vs nonV600E
```{r}
MTAB_metadata_v600e_vs_nonv600e <- metadata_MTAB_mutation_filtered %>% filter(!(metadata_MTAB_mutation_filtered$mutation == "wt") | (metadata_MTAB_mutation_filtered$mutation == "p.Val640Glu"))

MTAB_metadata_v600e_vs_nonv600e$mutation[MTAB_metadata_v600e_vs_nonv600e$mutation != "p.Val640Glu"] <- "nonV600E"

MTAB_counts_diffexp <- MTAB_counts
colnames(MTAB_counts_diffexp) <- gsub("\\.", "-", colnames(MTAB_counts_diffexp))
rownames(MTAB_counts_diffexp) <- MTAB_counts_diffexp$`ENSG-ID`
MTAB_counts_diffexp <- MTAB_counts_diffexp[, -1]

MTAB_counts_filtered_v600e_vs_nonv600e <- MTAB_counts_diffexp[, colnames(MTAB_counts_diffexp) %in% MTAB_metadata_v600e_vs_nonv600e$DNA_Tumor_Sample_Barcode]
```

```{r}
colnames(MTAB_metadata_v600e_vs_nonv600e)[40] <- "MSI_status"
colnames(MTAB_metadata_v600e_vs_nonv600e)[33] <- "Tumour_Cell_Content_Pathology"
colnames(MTAB_metadata_v600e_vs_nonv600e)[17] <- "Age_at_diagnosis"
colnames(MTAB_metadata_v600e_vs_nonv600e)[21] <- "Primary_Site_Disease"
colnames(MTAB_metadata_v600e_vs_nonv600e)[23] <- "Tumour_Site"

rownames(MTAB_metadata_v600e_vs_nonv600e) <- MTAB_metadata_v600e_vs_nonv600e$DNA_Tumor_Sample_Barcode
remove_col_mtab_v600e_vs_nonv600e <- setdiff(rownames(MTAB_metadata_v600e_vs_nonv600e), colnames(MTAB_counts_filtered_v600e_vs_nonv600e))

MTAB_metadata_v600e_vs_nonv600e <- MTAB_metadata_v600e_vs_nonv600e %>% filter(!(MTAB_metadata_v600e_vs_nonv600e$DNA_Tumor_Sample_Barcode %in% remove_col_mtab_v600e_vs_nonv600e))

MTAB_metadata_v600e_vs_nonv600e$Sex <- factor(MTAB_metadata_v600e_vs_nonv600e$Sex)
MTAB_metadata_v600e_vs_nonv600e$MSI_status <- factor(MTAB_metadata_v600e_vs_nonv600e$MSI_status)
MTAB_metadata_v600e_vs_nonv600e$Tumour_Cell_Content_Pathology <- as.numeric(MTAB_metadata_v600e_vs_nonv600e$Tumour_Cell_Content_Pathology)
MTAB_metadata_v600e_vs_nonv600e$Tumour_Site <- factor(MTAB_metadata_v600e_vs_nonv600e$Tumour_Site)
MTAB_metadata_v600e_vs_nonv600e$mutation <- factor(MTAB_metadata_v600e_vs_nonv600e$mutation)


dds_MTAB_diffexp_v600e_vs_nonv600e <- DESeqDataSetFromMatrix(countData = MTAB_counts_filtered_v600e_vs_nonv600e,
                                   colData = MTAB_metadata_v600e_vs_nonv600e,
                                   design = ~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation)

save(dds_MTAB_diffexp_v600e_vs_nonv600e, MTAB_counts_filtered_v600e_vs_nonv600e, MTAB_metadata_v600e_vs_nonv600e, file = "DESeq2_MTAB_V600E_vs_nonV600E.RData")

# load("DESeq2_MTAB_V600E_vs_nonV600E.RData")
# 
# keep <- rowSums(counts(dds_MTAB_diffexp_v600e_vs_nonv600e) >= 10) >= 28
# dds_MTAB_diffexp_v600e_vs_nonv600e <- dds_MTAB_diffexp_v600e_vs_nonv600e[keep,]
#  
# dds_MTAB_diffexp_v600e_vs_nonv600e <- DESeq(dds_MTAB_diffexp_v600e_vs_nonv600e)
# save(dds_MTAB_diffexp_v600e_vs_nonv600e, file = "DESeq2_MTAB_V600E_vs_nonV600E_results.RData")

```


```{r}
load("DESeq2_MTAB_V600E_vs_nonV600E_results.RData")

dds_MTAB_diffexp_v600e_vs_nonv600e
res_dds_MTAB_diffexp_v600e_vs_nonv600e <- results(dds_MTAB_diffexp_v600e_vs_nonv600e, alpha = 0.05, contrast = c('mutation', 'p.Val640Glu','nonV600E'))
res_dds_MTAB_diffexp_v600e_vs_nonv600e <- res_dds_MTAB_diffexp_v600e_vs_nonv600e[order(res_dds_MTAB_diffexp_v600e_vs_nonv600e$padj), ]
res_dds_MTAB_diffexp_v600e_vs_nonv600e_df <- as.data.frame(res_dds_MTAB_diffexp_v600e_vs_nonv600e)
res_dds_MTAB_diffexp_v600e_vs_nonv600e_df$Gene <- rownames(res_dds_MTAB_diffexp_v600e_vs_nonv600e_df)

res_dds_MTAB_diffexp_v600e_vs_nonv600e_df <- res_dds_MTAB_diffexp_v600e_vs_nonv600e_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_dds_MTAB_diffexp_v600e_vs_nonv600e_df, "DESeq2_MTAB_V600E_vs_nonV600E_results.csv", row.names = FALSE)
```

```{r}
DEG_MTAB_V600E_VS_nonV600E <- read.csv("DESeq2_MTAB_V600E_vs_nonV600E_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_MTAB_V600E_VS_nonV600E <- DEG_MTAB_V600E_VS_nonV600E %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_MTAB_V600E_VS_nonV600E <- DEG_MTAB_V600E_VS_nonV600E %>% filter(!is.na(log2FoldChange))
DEG_MTAB_V600E_VS_nonV600E <- DEG_MTAB_V600E_VS_nonV600E %>% filter(!is.na(gene_symbol))


ranked_genes_mtab_v600e_vs_nonv600e <- DEG_MTAB_V600E_VS_nonV600E$log2FoldChange
names(ranked_genes_mtab_v600e_vs_nonv600e) <- DEG_MTAB_V600E_VS_nonV600E$gene_symbol

ranked_genes_mtab_v600e_vs_nonv600e <- sort(ranked_genes_mtab_v600e_vs_nonv600e, decreasing=TRUE)
ranked_genes_mtab_v600e_vs_nonv600e <- ranked_genes_mtab_v600e_vs_nonv600e[!duplicated(names(ranked_genes_mtab_v600e_vs_nonv600e))]
```

### VOLCANO
```{r}
library(EnhancedVolcano)

MTAB_v600e_vs_nonv600e_volcano <- EnhancedVolcano(DEG_MTAB_V600E_VS_nonV600E,
                 lab = DEG_MTAB_V600E_VS_nonV600E$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "MTAB BRAF V600E vs nonV600E",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
MTAB_v600e_vs_nonv600e_volcano
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_mtab_v600e_vs_nonv600e <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_mtab_v600e_vs_nonv600e) #not specify permutations
```

```{r}
fgsea_results_mtab_v600e_vs_nonv600e <- fgsea_results_mtab_v600e_vs_nonv600e[order(fgsea_results_mtab_v600e_vs_nonv600e$padj), ]

head(fgsea_results_mtab_v600e_vs_nonv600e %>% arrange(padj), n=15)
```

### HALLMARKS
```{r}
plot_mtab_v600e_nonv600e <- fgsea_results_mtab_v600e_vs_nonv600e %>% filter(pval < 0.05)
hallmarks_results_mtab_v600e_nonv600e <- ggplot(plot_mtab_v600e_nonv600e, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAF V600E vs nonV600E", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()

hallmarks_results_mtab_v600e_nonv600e
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_MTAB_v600e_vs_nonv600e_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_mtab_v600e_vs_nonv600e)
```


```{r}
fgsea_results_MTAB_v600e_vs_nonv600e_kegg <- fgsea_results_MTAB_v600e_vs_nonv600e_kegg[order(fgsea_results_MTAB_v600e_vs_nonv600e_kegg$padj), ]
```

### KEGG
```{r}
fgsea_results_MTAB_v600e_vs_nonv600e_kegg_PLOT <- fgsea_results_MTAB_v600e_vs_nonv600e_kegg %>% filter(pval < 0.05)
ggplot(fgsea_results_MTAB_v600e_vs_nonv600e_kegg_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAFV600E vs nonV600E KEGG", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```

```{r}
msigdb_reactome <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:REACTOME")
reactome_pathway_list <- split(msigdb_reactome$gene_symbol, msigdb_reactome$gs_name)
```


```{r}
fgsea_results_mtab_v600e_vs_nonv600e_reactome <- fgsea(pathways = reactome_pathway_list, 
                                            stats = ranked_genes_mtab_v600e_vs_nonv600e)
```


```{r}
fgsea_results_mtab_v600e_vs_nonv600e_reactome <- fgsea_results_mtab_v600e_vs_nonv600e_reactome[order(fgsea_results_mtab_v600e_vs_nonv600e_reactome$padj), ]
```
### REACTOME
```{r}
fgsea_results_mtab_v600e_vs_nonv600e_reactome_PLOT <- fgsea_results_mtab_v600e_vs_nonv600e_reactome %>% filter(padj < 0.05)
ggplot(fgsea_results_mtab_v600e_vs_nonv600e_reactome_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title=" MTAB BRAF V600E VS nonV600E REACTOME", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
```









# 1 vs 2 Comparisons
## TCGA
### V600E vs wt + nonV600E
```{r}
diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e <- final_diff_exp_TCGA_metadata
  
diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e$condition[!(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e$condition == "BRAFV600E")] <- "nonV600E_wt"

diff_exp_TCGA_counts_V600E_VS_wt_nonv600e <- final_diff_exp_TCGA_counts[, colnames(final_diff_exp_TCGA_counts) %in% rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e)]
```

```{r}
diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e <- diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e[
  rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e) %in% colnames(diff_exp_TCGA_counts_V600E_VS_wt_nonv600e), 
]

# Subset count matrix to include only samples present in the metadata
diff_exp_TCGA_counts_V600E_VS_wt_nonv600e <- diff_exp_TCGA_counts_V600E_VS_wt_nonv600e[, 
  colnames(diff_exp_TCGA_counts_V600E_VS_wt_nonv600e) %in% rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e)]

diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e <- diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e[
  match(colnames(diff_exp_TCGA_counts_V600E_VS_wt_nonv600e), rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e)), 
]

# Check if everything is aligned correctly
all(colnames(diff_exp_TCGA_counts_V600E_VS_wt_nonv600e) == rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e))
```

```{r}
diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e <- diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e %>%
  filter(!is.na(gender) & !is.na(MsiStatus) & !is.na(TumorPurity) & 
         !is.na(age_at_initial_pathologic_diagnosis) & !is.na(Site))

# Subset the counts matrix to match the filtered metadata
diff_exp_TCGA_counts_V600E_VS_wt_nonv600e <- diff_exp_TCGA_counts_V600E_VS_wt_nonv600e[, 
  colnames(diff_exp_TCGA_counts_V600E_VS_wt_nonv600e) %in% rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e)]

# Ensure correct order
diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e <- diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e[
  match(colnames(diff_exp_TCGA_counts_V600E_VS_wt_nonv600e), rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e)), ]


all(colnames(diff_exp_TCGA_counts_V600E_VS_wt_nonv600e) == rownames(diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e))
```


```{r}
dds_TCGA_v600e_vs_nonv600e_wt <- DESeqDataSetFromMatrix(countData = diff_exp_TCGA_counts_V600E_VS_wt_nonv600e,
                                   colData = diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e,
                                   design = ~ gender + MsiStatus + TumorPurity + age_at_initial_pathologic_diagnosis + Site + condition)

# save(dds_TCGA_v600e_vs_nonv600e_wt, diff_exp_TCGA_counts_V600E_VS_wt_nonv600e, diff_exp_TCGA_metadata_V600E_VS_wt_nonv600e, file = "DESeq2_TCGA_v600e_vs_nonv600e_wt.Rdata")

# load("DESeq2_TCGA_v600e_vs_nonv600e_wt.Rdata")
# 
# keep <- rowSums(counts(dds_TCGA_v600e_vs_nonv600e_wt) >= 10) >= 45
# dds_TCGA_v600e_vs_nonv600e_wt <- dds_TCGA_v600e_vs_nonv600e_wt[keep,]
#  
# dds_TCGA_v600e_vs_nonv600e_wt <- DESeq(dds_TCGA_v600e_vs_nonv600e_wt)
# save(dds_TCGA_v600e_vs_nonv600e_wt, file = "DESeq2_TCGA_v600e_vs_nonv600e_wt_RESULTS.Rdata")
```


```{r}
load("DESeq2_TCGA_v600e_vs_nonv600e_wt_RESULTS.Rdata")

dds_TCGA_v600e_vs_nonv600e_wt
res_v600e_vs_nonv600e_wt <- results(dds_TCGA_v600e_vs_nonv600e_wt, alpha = 0.05, contrast = c('condition', 'BRAFV600E','nonV600E_wt'))
res_v600e_vs_nonv600e_wt <- res_v600e_vs_nonv600e_wt[order(res_v600e_vs_nonv600e_wt$padj), ]
res_v600e_vs_nonv600e_wt_df <- as.data.frame(res_v600e_vs_nonv600e_wt)
res_v600e_vs_nonv600e_wt_df$Gene <- rownames(res_v600e_vs_nonv600e_wt_df)

res_v600e_vs_nonv600e_wt_df <- res_v600e_vs_nonv600e_wt_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_v600e_vs_nonv600e_wt_df, "DESeq2_TCGA_v600e_vs_nonv600e_wt_results.csv", row.names = FALSE)

```

```{r}
DEG_V600E_vs_nonV600E_wt <- read.csv("DESeq2_TCGA_v600e_vs_nonv600e_wt_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_V600E_vs_nonV600E_wt <- DEG_V600E_vs_nonV600E_wt %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_V600E_vs_nonV600E_wt <- DEG_V600E_vs_nonV600E_wt %>% filter(!is.na(log2FoldChange))
DEG_V600E_vs_nonV600E_wt <- DEG_V600E_vs_nonV600E_wt %>% filter(!is.na(gene_symbol))
# DEG_wt_vs_V600E <- DEG_wt_vs_V600E %>% filter(DEG_wt_vs_V600E$pvalue < 0.05)#get rid of the
ranked_genes_V600E_vs_nonV600E_wt <- DEG_V600E_vs_nonV600E_wt$log2FoldChange
names(ranked_genes_V600E_vs_nonV600E_wt) <- DEG_V600E_vs_nonV600E_wt$gene_symbol

ranked_genes_V600E_vs_nonV600E_wt <- sort(ranked_genes_V600E_vs_nonV600E_wt, decreasing=TRUE)
ranked_genes_V600E_vs_nonV600E_wt <- ranked_genes_V600E_vs_nonV600E_wt[!duplicated(names(ranked_genes_V600E_vs_nonV600E_wt))]
```

####VOLCANO
```{r}
library(EnhancedVolcano)

v600e_vs_nonv600e_wt <- EnhancedVolcano(DEG_V600E_vs_nonV600E_wt,
                 lab = DEG_V600E_vs_nonV600E_wt$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "BRAF V600E vs wt_v600e",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
v600e_vs_nonv600e_wt
```


```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_tcga_V600E_vs_nonV600E_wt <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_V600E_vs_nonV600E_wt) #not specify permutations
```

```{r}
fgsea_results_tcga_V600E_vs_nonV600E_wt <- fgsea_results_tcga_V600E_vs_nonV600E_wt[order(fgsea_results_tcga_V600E_vs_nonV600E_wt$padj), ]

head(fgsea_results_tcga_V600E_vs_nonV600E_wt %>% arrange(padj), n=15)
```
####HALLMARKS
```{r}
plot_TCGA_V600E_vs_nonV600E_wt <- fgsea_results_tcga_V600E_vs_nonV600E_wt %>% filter(pval < 0.05)
hallmarks_TCGA_V600E_vs_nonV600E_wt <- ggplot(plot_TCGA_V600E_vs_nonV600E_wt, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="TCGA BRAF V600E vs WT+nonV600E", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()

hallmarks_TCGA_V600E_vs_nonV600E_wt
```


```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_MTAB_v600e_vs_nonv600_wt_e_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_V600E_vs_nonV600E_wt)
```


```{r}
fgsea_results_MTAB_v600e_vs_nonv600_wt_e_kegg <- fgsea_results_MTAB_v600e_vs_nonv600_wt_e_kegg[order(fgsea_results_MTAB_v600e_vs_nonv600_wt_e_kegg$padj), ]
```

#### KEGG
```{r}
fgsea_results_MTAB_v600e_vs_nonv600e_wt_kegg_PLOT <- fgsea_results_MTAB_v600e_vs_nonv600_wt_e_kegg %>% filter(pval < 0.01)
tcga_combo_v600e_vs_non_wt <- ggplot(fgsea_results_MTAB_v600e_vs_nonv600e_wt_kegg_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAFV600E vs nonV600E KEGG", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
tcga_combo_v600e_vs_non_wt
```


























### nonV600E vs wt + V600E
```{r}
diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e <- final_diff_exp_TCGA_metadata
  
diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e$condition[!(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e$condition == "BRAFnonV600E")] <- "V600E_wt"

diff_exp_TCGA_counts_nonV600E_VS_wt_v600e <- final_diff_exp_TCGA_counts[, colnames(final_diff_exp_TCGA_counts) %in% rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e)]
```

```{r}
diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e <- diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e[
  rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e) %in% colnames(diff_exp_TCGA_counts_nonV600E_VS_wt_v600e), 
]

# Subset count matrix to include only samples present in the metadata
diff_exp_TCGA_counts_nonV600E_VS_wt_v600e <- diff_exp_TCGA_counts_nonV600E_VS_wt_v600e[, 
  colnames(diff_exp_TCGA_counts_nonV600E_VS_wt_v600e) %in% rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e)]

diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e <- diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e[
  match(colnames(diff_exp_TCGA_counts_nonV600E_VS_wt_v600e), rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e)), 
]

# Check if everything is aligned correctly
all(colnames(diff_exp_TCGA_counts_nonV600E_VS_wt_v600e) == rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e))
```

```{r}
diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e <- diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e %>%
  filter(!is.na(gender) & !is.na(MsiStatus) & !is.na(TumorPurity) & 
         !is.na(age_at_initial_pathologic_diagnosis) & !is.na(Site))

# Subset the counts matrix to match the filtered metadata
diff_exp_TCGA_counts_nonV600E_VS_wt_v600e <- diff_exp_TCGA_counts_nonV600E_VS_wt_v600e[, 
  colnames(diff_exp_TCGA_counts_nonV600E_VS_wt_v600e) %in% rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e)]

# Ensure correct order
diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e <- diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e[
  match(colnames(diff_exp_TCGA_counts_nonV600E_VS_wt_v600e), rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e)), ]


all(colnames(diff_exp_TCGA_counts_nonV600E_VS_wt_v600e) == rownames(diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e))
```


```{r}
dds_TCGA_nonv600e_vs_v600e_wt <- DESeqDataSetFromMatrix(countData = diff_exp_TCGA_counts_nonV600E_VS_wt_v600e,
                                   colData = diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e,
                                   design = ~ gender + MsiStatus + TumorPurity + age_at_initial_pathologic_diagnosis + Site + condition)

save(dds_TCGA_nonv600e_vs_v600e_wt, diff_exp_TCGA_counts_nonV600E_VS_wt_v600e, diff_exp_TCGA_metadata_nonV600E_VS_wt_v600e, file = "DESeq2_TCGA_nonv600e_vs_v600e_wt.Rdata")
```










```{r}
load("DESeq2_TCGA_nonv600e_vs_v600e_wt_RESULTS.Rdata")

dds_TCGA_nonv600e_vs_v600e_wt
res_nonv600e_vs_v600e_wt <- results(dds_TCGA_nonv600e_vs_v600e_wt, alpha = 0.05, contrast = c('condition', 'BRAFnonV600E','V600E_wt'))
res_nonv600e_vs_v600e_wt <- res_nonv600e_vs_v600e_wt[order(res_nonv600e_vs_v600e_wt$padj), ]
res_nonv600e_vs_v600e_wt_df <- as.data.frame(res_nonv600e_vs_v600e_wt)
res_nonv600e_vs_v600e_wt_df$Gene <- rownames(res_nonv600e_vs_v600e_wt_df)

res_nonv600e_vs_v600e_wt_df <- res_nonv600e_vs_v600e_wt_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_nonv600e_vs_v600e_wt_df, "DESeq2_TCGA_nonv600e_vs_v600e_wt_results.csv", row.names = FALSE)

```

```{r}
DEG_nonV600E_vs_V600E_wt <- read.csv("DESeq2_TCGA_nonv600e_vs_v600e_wt_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_nonV600E_vs_V600E_wt <- DEG_nonV600E_vs_V600E_wt %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_nonV600E_vs_V600E_wt <- DEG_nonV600E_vs_V600E_wt %>% filter(!is.na(log2FoldChange))
DEG_nonV600E_vs_V600E_wt <- DEG_nonV600E_vs_V600E_wt %>% filter(!is.na(gene_symbol))
# DEG_wt_vs_V600E <- DEG_wt_vs_V600E %>% filter(DEG_wt_vs_V600E$pvalue < 0.05)#get rid of the
ranked_genes_nonV600E_vs_V600E_wt <- DEG_nonV600E_vs_V600E_wt$log2FoldChange
names(ranked_genes_nonV600E_vs_V600E_wt) <- DEG_nonV600E_vs_V600E_wt$gene_symbol

ranked_genes_nonV600E_vs_V600E_wt <- sort(ranked_genes_nonV600E_vs_V600E_wt, decreasing=TRUE)
ranked_genes_nonV600E_vs_V600E_wt <- ranked_genes_nonV600E_vs_V600E_wt[!duplicated(names(ranked_genes_nonV600E_vs_V600E_wt))]
```

####VOLCANO
```{r}
library(EnhancedVolcano)

nonv600e_vs_v600e_wt <- EnhancedVolcano(DEG_nonV600E_vs_V600E_wt,
                 lab = DEG_nonV600E_vs_V600E_wt$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "BRAF nonV600E vs wt_v600e",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
nonv600e_vs_v600e_wt
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_tcga_nonV600E_vs_V600E_wt <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_nonV600E_vs_V600E_wt) #not specify permutations
```

```{r}
fgsea_results_tcga_nonV600E_vs_V600E_wt <- fgsea_results_tcga_nonV600E_vs_V600E_wt[order(fgsea_results_tcga_nonV600E_vs_V600E_wt$padj), ]

head(fgsea_results_tcga_nonV600E_vs_V600E_wt %>% arrange(padj), n=15)
```
####HALLMARKS
```{r}
plot_TCGA_nonV600E_vs_V600E_wt <- fgsea_results_tcga_nonV600E_vs_V600E_wt %>% filter(pval < 0.05)
hallmarks_TCGA_nonV600E_vs_V600E_wt <- ggplot(plot_TCGA_nonV600E_vs_V600E_wt, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="TCGA BRAF nonV600E vs WT+V600E", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()

hallmarks_TCGA_nonV600E_vs_V600E_wt
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_MTAB_nonv600e_vs_v600_wt_e_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_nonV600E_vs_V600E_wt)
```


```{r}
fgsea_results_MTAB_nonv600e_vs_v600_wt_e_kegg <- fgsea_results_MTAB_nonv600e_vs_v600_wt_e_kegg[order(fgsea_results_MTAB_nonv600e_vs_v600_wt_e_kegg$padj), ]
```

#### KEGG
```{r}
fgsea_results_MTAB_nonv600e_vs_v600_wt_e_kegg_PLOT <- fgsea_results_MTAB_nonv600e_vs_v600_wt_e_kegg %>% filter(pval < 0.05)
tcga_combo_nonv600e_vs_v600e_wt <- ggplot(fgsea_results_MTAB_nonv600e_vs_v600_wt_e_kegg_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB BRAFnonV600E vs V600E_wt KEGG", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
tcga_combo_nonv600e_vs_v600e_wt
```








### wt vs nonV600E + V600E
```{r}
diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e <- final_diff_exp_TCGA_metadata
  
diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e$condition[!(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e$condition == "wt")] <- "V600E_nonV600E"

diff_exp_TCGA_counts_wt_VS_v600e_nonv600e <- final_diff_exp_TCGA_counts[, colnames(final_diff_exp_TCGA_counts) %in% rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e)]
```

```{r}
diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e <- diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e[
  rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e) %in% colnames(diff_exp_TCGA_counts_wt_VS_v600e_nonv600e), 
]

# Subset count matrix to include only samples present in the metadata
diff_exp_TCGA_counts_wt_VS_v600e_nonv600e <- diff_exp_TCGA_counts_wt_VS_v600e_nonv600e[, 
  colnames(diff_exp_TCGA_counts_wt_VS_v600e_nonv600e) %in% rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e)]

diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e <- diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e[
  match(colnames(diff_exp_TCGA_counts_wt_VS_v600e_nonv600e), rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e)), 
]

# Check if everything is aligned correctly
all(colnames(diff_exp_TCGA_counts_wt_VS_v600e_nonv600e) == rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e))
```

```{r}
diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e <- diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e %>%
  filter(!is.na(gender) & !is.na(MsiStatus) & !is.na(TumorPurity) & 
         !is.na(age_at_initial_pathologic_diagnosis) & !is.na(Site))

# Subset the counts matrix to match the filtered metadata
diff_exp_TCGA_counts_wt_VS_v600e_nonv600e <- diff_exp_TCGA_counts_wt_VS_v600e_nonv600e[, 
  colnames(diff_exp_TCGA_counts_wt_VS_v600e_nonv600e) %in% rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e)]

# Ensure correct order
diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e <- diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e[
  match(colnames(diff_exp_TCGA_counts_wt_VS_v600e_nonv600e), rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e)), ]


all(colnames(diff_exp_TCGA_counts_wt_VS_v600e_nonv600e) == rownames(diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e))
```


```{r}
dds_TCGA_wt_vs_v600e_nonv600e <- DESeqDataSetFromMatrix(countData = diff_exp_TCGA_counts_wt_VS_v600e_nonv600e,
                                   colData = diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e,
                                   design = ~ gender + MsiStatus + TumorPurity + age_at_initial_pathologic_diagnosis + Site + condition)

save(dds_TCGA_wt_vs_v600e_nonv600e, diff_exp_TCGA_counts_wt_VS_v600e_nonv600e, diff_exp_TCGA_metadata_wt_vs_v600e_nonv600e, file = "DESeq2_TCGA_wt_vs_v600e_nonv600e.Rdata")
```











```{r}
load("DESeq2_TCGA_wt_vs_v600e_nonv600e_RESULTS.Rdata")

dds_TCGA_wt_vs_v600e_nonv600e
res_wt_vs_v600e_nonv600e <- results(dds_TCGA_wt_vs_v600e_nonv600e, alpha = 0.05, contrast = c('condition', 'wt','V600E_nonV600E'))
res_wt_vs_v600e_nonv600e <- res_wt_vs_v600e_nonv600e[order(res_wt_vs_v600e_nonv600e$padj), ]
res_wt_vs_v600e_nonv600e_df <- as.data.frame(res_wt_vs_v600e_nonv600e)
res_wt_vs_v600e_nonv600e_df$Gene <- rownames(res_wt_vs_v600e_nonv600e_df)

res_wt_vs_v600e_nonv600e_df <- res_wt_vs_v600e_nonv600e_df %>%
  mutate(Gene = sub("\\..*", "", Gene))

write.csv(res_wt_vs_v600e_nonv600e_df, "DESeq2_TCGA_wt_vs_v600e_nonv600e_results.csv", row.names = FALSE)

```

```{r}
DEG_wt_vs_V600E_nonV600E <- read.csv("DESeq2_TCGA_wt_vs_v600e_nonv600e_results.csv")
library(fgsea)
library(msigdbr)
library(org.Hs.eg.db)
```

```{r}
DEG_wt_vs_V600E_nonV600E <- DEG_wt_vs_V600E_nonV600E %>%
    mutate(gene_symbol = mapIds(org.Hs.eg.db, 
                                keys = Gene, 
                                column = "SYMBOL", 
                                keytype = "ENSEMBL",
                                multiVals = "first"))
```

```{r}
DEG_wt_vs_V600E_nonV600E <- DEG_wt_vs_V600E_nonV600E %>% filter(!is.na(log2FoldChange))
DEG_wt_vs_V600E_nonV600E <- DEG_wt_vs_V600E_nonV600E %>% filter(!is.na(gene_symbol))
# DEG_wt_vs_V600E <- DEG_wt_vs_V600E %>% filter(DEG_wt_vs_V600E$pvalue < 0.05)#get rid of the
ranked_genes_wt_vs_V600E_nonV600E <- DEG_wt_vs_V600E_nonV600E$log2FoldChange
names(ranked_genes_wt_vs_V600E_nonV600E) <- DEG_wt_vs_V600E_nonV600E$gene_symbol

ranked_genes_wt_vs_V600E_nonV600E <- sort(ranked_genes_wt_vs_V600E_nonV600E, decreasing=TRUE)
ranked_genes_wt_vs_V600E_nonV600E <- ranked_genes_wt_vs_V600E_nonV600E[!duplicated(names(ranked_genes_wt_vs_V600E_nonV600E))]
```

####VOLCANO
```{r}
library(EnhancedVolcano)

nonv600e_vs_v600e_wt <- EnhancedVolcano(DEG_wt_vs_V600E_nonV600E,
                 lab = DEG_wt_vs_V600E_nonV600E$gene_symbol,
                 x = "log2FoldChange",
                 y = "pvalue",
                 pCutoff = 0.05,
                 FCcutoff = 1,
                 title = "BRAF wt vs V600E_nonV600E",
                 legendLabels = c("No sig", "Log2 FC", "p-value", "Both"),
                 pointSize = 2.0,
                 labSize = 3.0)
nonv600e_vs_v600e_wt
```

```{r}
msigdb_pathways <- msigdbr(species = "Homo sapiens", category = "H")
pathway_list <- split(msigdb_pathways$gene_symbol, msigdb_pathways$gs_name)
```

```{r}
fgsea_results_tcga_wt_vs_nonV600E_v600e <- fgsea(pathways = pathway_list, 
                       stats = ranked_genes_wt_vs_V600E_nonV600E) #not specify permutations
```

```{r}
fgsea_results_tcga_wt_vs_nonV600E_v600e <- fgsea_results_tcga_wt_vs_nonV600E_v600e[order(fgsea_results_tcga_wt_vs_nonV600E_v600e$padj), ]

head(fgsea_results_tcga_wt_vs_nonV600E_v600e %>% arrange(padj), n=15)
```
####HALLMARKS
```{r}
plot_TCGA_wt_vs_V600E_nonv600e <- fgsea_results_tcga_wt_vs_nonV600E_v600e %>% filter(pval < 0.05)
hallmarks_TCGA_wt_vs_nonV600E_v600e <- ggplot(plot_TCGA_wt_vs_V600E_nonv600e, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="TCGA BRAF wt vs nonV600E+V600E", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()

hallmarks_TCGA_wt_vs_nonV600E_v600e
```

```{r}
msigdb_kegg <- msigdbr(species = "Homo sapiens", category = "C2", subcategory = "CP:KEGG")
kegg_pathway_list <- split(msigdb_kegg$gene_symbol, msigdb_kegg$gs_name)
```


```{r}
fgsea_results_MTAB_wt_vs_v600_nonv600e_e_kegg <- fgsea(pathways = kegg_pathway_list, 
                                        stats = ranked_genes_wt_vs_V600E_nonV600E)
```


```{r}
fgsea_results_MTAB_wt_vs_v600_nonv600e_e_kegg <- fgsea_results_MTAB_wt_vs_v600_nonv600e_e_kegg[order(fgsea_results_MTAB_wt_vs_v600_nonv600e_e_kegg$padj), ]
```

#### KEGG
```{r}
fgsea_results_MTAB_wt_vs_v600_nonv600e_e_kegg_PLOT <- fgsea_results_MTAB_wt_vs_v600_nonv600e_e_kegg %>% filter(pval < 0.05)
tcga_combo_wt_vs_v600e_nonv600e <- ggplot(fgsea_results_MTAB_wt_vs_v600_nonv600e_e_kegg_PLOT, aes(reorder(pathway, NES), NES)) + 
  geom_col(aes(fill=NES > 0)) + 
  coord_flip() + 
  labs(title="MTAB wy vs V600E_nonv600e KEGG", x="Pathway", y="Normalized Enrichment Score") +
  theme_minimal()
tcga_combo_wt_vs_v600e_nonv600e
```












##MTAB
### V600E vs wt + nonV600E
```{r}
metadata_MTAB_v600e_vs_wt_nonv600e <- metadata_MTAB_mutation %>% filter(!metadata_MTAB_mutation$`Anatomic Organ Subdivision` == "Rectum")
metadata_MTAB_v600e_vs_wt_nonv600e <- metadata_MTAB_v600e_vs_wt_nonv600e[,-1]
vector_mut <- c("p.Val640Glu")
metadata_MTAB_v600e_vs_wt_nonv600e$mutation[!(metadata_MTAB_v600e_vs_wt_nonv600e$mutation %in% vector_mut)] <- "nonV600E_wt"
```

```{r}
MTAB_counts_v600e_vs_wt_nonv600e <- MTAB_counts
colnames(MTAB_counts_v600e_vs_wt_nonv600e) <- gsub("\\.", "-", colnames(MTAB_counts_v600e_vs_wt_nonv600e))
rownames(MTAB_counts_v600e_vs_wt_nonv600e) <- MTAB_counts_v600e_vs_wt_nonv600e$`ENSG-ID`
MTAB_counts_v600e_vs_wt_nonv600e <- MTAB_counts_v600e_vs_wt_nonv600e[, -1]

MTAB_counts_v600e_vs_wt_nonv600e <- MTAB_counts_v600e_vs_wt_nonv600e[, colnames(MTAB_counts_v600e_vs_wt_nonv600e) %in% metadata_MTAB_v600e_vs_wt_nonv600e$DNA_Tumor_Sample_Barcode]

metadata_MTAB_v600e_vs_wt_nonv600e <- metadata_MTAB_v600e_vs_wt_nonv600e[
  metadata_MTAB_v600e_vs_wt_nonv600e$DNA_Tumor_Sample_Barcode %in% colnames(MTAB_counts_v600e_vs_wt_nonv600e),]
```


```{r}
colnames(metadata_MTAB_v600e_vs_wt_nonv600e)[40] <- "MSI_status"
colnames(metadata_MTAB_v600e_vs_wt_nonv600e)[33] <- "Tumour_Cell_Content_Pathology"
colnames(metadata_MTAB_v600e_vs_wt_nonv600e)[17] <- "Age_at_diagnosis"
colnames(metadata_MTAB_v600e_vs_wt_nonv600e)[21] <- "Primary_Site_Disease"
colnames(metadata_MTAB_v600e_vs_wt_nonv600e)[23] <- "Tumour_Site"
```

```{r}
dds_MTAB_v600e_vs_wt_nonv600e <- DESeqDataSetFromMatrix(countData = MTAB_counts_v600e_vs_wt_nonv600e,
                                   colData = metadata_MTAB_v600e_vs_wt_nonv600e,
                                   design = ~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation)

# save(dds_MTAB_v600e_vs_wt_nonv600e, MTAB_counts_v600e_vs_wt_nonv600e, metadata_MTAB_v600e_vs_wt_nonv600e, file = "DESeq2_MTAB_v600e_vs_wt_nonv600e.Rdata")
```




















#limma voom
## v600e vs wt_nonv600e
```{r}
library(edgeR)
library(limma)
```
create dge_list
```{r}
dge <- DGEList(counts = MTAB_counts_v600e_vs_wt_nonv600e)
```

```{r}
dge <- calcNormFactors(dge)
dim(dge)
```

```{r}
cutoff <- 18
drop <- which(apply(cpm(dge), 1, max) < cutoff)
dge_drop <- dge[-drop,] 
dim(dge_drop)
```


```{r}
design <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation, data = metadata_MTAB_v600e_vs_wt_nonv600e)

```

```{r}
v <- voom(dge_drop, design, plot = TRUE)
```

```{r}
fit <- lmFit(v, design)
fit <- eBayes(fit)
```

```{r}
topTable_v600e <- topTable(fit,
                           coef = "mutationp.Val640Glu",
                           adjust.method = "BH",
                           number = Inf)
topTable_v600e$Gene <- rownames(topTable_v600e)
```

```{r}
topTable_v600e$Gene <- mapIds(org.Hs.eg.db,
                              keys = rownames(topTable_v600e),
                              column = "SYMBOL",
                              keytype = "ENSEMBL",
                              multiVals = "first")
topTable_v600e <- topTable_v600e[!is.na(topTable_v600e$Gene), ]
topTable_v600e <- topTable_v600e[!duplicated(topTable_v600e$Gene), ]
```


```{r}
EnhancedVolcano(topTable_v600e,
                lab = topTable_v600e$Gene,
                x = "logFC",
                y = "P.Value",
                pCutoff = 0.05,
                FCcutoff = 1,
                title = "Limma-Voom: BRAFV600E vs nonV600E/wt",
                pointSize = 2.0,
                labSize = 3.0)
```













## nonv600e vs wt_v600e
```{r}
metadata_MTAB_nonv600e_vs_wt_v600e <- metadata_MTAB_mutation %>% filter(!metadata_MTAB_mutation$`Anatomic Organ Subdivision` == "Rectum")
metadata_MTAB_nonv600e_vs_wt_v600e <- metadata_MTAB_nonv600e_vs_wt_v600e[,-1]
vector_mut_nonv600e <- c("p.Val640Glu", "wt")
metadata_MTAB_nonv600e_vs_wt_v600e$mutation[!(metadata_MTAB_nonv600e_vs_wt_v600e$mutation %in% vector_mut_nonv600e)] <- "nonV600E"
metadata_MTAB_nonv600e_vs_wt_v600e$mutation[(metadata_MTAB_nonv600e_vs_wt_v600e$mutation %in% vector_mut_nonv600e)] <- "V600E_wt"

metadata_MTAB_nonv600e_vs_wt_v600e$mutation <- factor(metadata_MTAB_nonv600e_vs_wt_v600e$mutation)
metadata_MTAB_nonv600e_vs_wt_v600e$mutation <- relevel(metadata_MTAB_nonv600e_vs_wt_v600e$mutation, ref = "V600E_wt")
```


```{r}
MTAB_counts_nonv600e_vs_wt_v600e <- MTAB_counts
rownames(MTAB_counts_nonv600e_vs_wt_v600e) <- MTAB_counts_nonv600e_vs_wt_v600e$ENSG.ID
MTAB_counts_nonv600e_vs_wt_v600e <- MTAB_counts_nonv600e_vs_wt_v600e[, -1]
```

```{r}
MTAB_counts_nonv600e_vs_wt_v600e <- MTAB_counts_nonv600e_vs_wt_v600e[, colnames(MTAB_counts_nonv600e_vs_wt_v600e) %in% metadata_MTAB_nonv600e_vs_wt_v600e$`RNA Tumor Sample Barcode`]

metadata_MTAB_nonv600e_vs_wt_v600e <- metadata_MTAB_nonv600e_vs_wt_v600e[
  metadata_MTAB_nonv600e_vs_wt_v600e$`RNA Tumor Sample Barcode` %in% colnames(MTAB_counts_nonv600e_vs_wt_v600e),]
```

```{r}
colnames(metadata_MTAB_nonv600e_vs_wt_v600e)[40] <- "MSI_status"
colnames(metadata_MTAB_nonv600e_vs_wt_v600e)[33] <- "Tumour_Cell_Content_Pathology"
colnames(metadata_MTAB_nonv600e_vs_wt_v600e)[17] <- "Age_at_diagnosis"
colnames(metadata_MTAB_nonv600e_vs_wt_v600e)[21] <- "Primary_Site_Disease"
colnames(metadata_MTAB_nonv600e_vs_wt_v600e)[23] <- "Tumour_Site"
```



```{r}
dge_nonv600e_vs_wt_v600e <- DGEList(counts = MTAB_counts_nonv600e_vs_wt_v600e)
```

```{r}
dge_nonv600e_vs_wt_v600e <- calcNormFactors(dge_nonv600e_vs_wt_v600e)
dim(dge_nonv600e_vs_wt_v600e)
```

```{r}
cutoff <- 1
drop_nonv600e_vs_wt_v600e <- which(apply(cpm(dge_nonv600e_vs_wt_v600e), 1, max) < cutoff)
dge_drop_nonv600e_vs_wt_v600e <- dge_nonv600e_vs_wt_v600e[-drop_nonv600e_vs_wt_v600e,] 
dim(dge_drop_nonv600e_vs_wt_v600e)
```


```{r}
design_nonv600e_vs_wt_v600e <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation, data = metadata_MTAB_nonv600e_vs_wt_v600e)

```

```{r}
v_nonv600e_vs_wt_v600e <- voom(dge_drop_nonv600e_vs_wt_v600e, design_nonv600e_vs_wt_v600e, plot = TRUE)
```


```{r}
fit_nonv600e_vs_wt_v600e <- lmFit(v_nonv600e_vs_wt_v600e, design_nonv600e_vs_wt_v600e)
fit_nonv600e_vs_wt_v600e <- eBayes(fit_nonv600e_vs_wt_v600e)
```

```{r}
topTable_nonv600e_vs_wt_v600e <- topTable(fit_nonv600e_vs_wt_v600e,
                           coef = "mutationnonV600E",
                           adjust.method = "BH",
                           number = Inf)
topTable_nonv600e_vs_wt_v600e$Gene <- rownames(topTable_nonv600e_vs_wt_v600e)
```

```{r}
topTable_nonv600e_vs_wt_v600e$Gene <- mapIds(org.Hs.eg.db,
                                             keys = rownames(topTable_nonv600e_vs_wt_v600e),
                                             column = "SYMBOL",
                                             keytype = "ENTREZID",
                                             multiVals = "first")


# topTable_nonv600e_vs_wt_v600e <- topTable_nonv600e_vs_wt_v600e[!is.na(topTable_nonv600e_vs_wt_v600e$Gene), ]
# topTable_nonv600e_vs_wt_v600e <- topTable_nonv600e_vs_wt_v600e[!duplicated(topTable_nonv600e_vs_wt_v600e$Gene), ]
```


```{r}
EnhancedVolcano(topTable_nonv600e_vs_wt_v600e,
                lab = topTable_nonv600e_vs_wt_v600e$Gene,
                x = "logFC",
                y = "P.Value",
                pCutoff = 0.05,
                FCcutoff = 1,
                title = "Limma-Voom: BRAFnonV600E vs V600E/wt",
                pointSize = 2.0,
                labSize = 3.0)
```


##wt vs nonv600e/v600e


```{r}
metadata_MTAB_wt_vs_v600e_nonv600e <- metadata_MTAB_mutation %>% filter(!metadata_MTAB_mutation$`Anatomic Organ Subdivision` == "Rectum")
metadata_MTAB_wt_vs_v600e_nonv600e <- metadata_MTAB_wt_vs_v600e_nonv600e[,-1]
vector_mut_wt_vs_v600e_nonv600e <- c("wt")
metadata_MTAB_wt_vs_v600e_nonv600e$mutation[!(metadata_MTAB_wt_vs_v600e_nonv600e$mutation %in% vector_mut_wt_vs_v600e_nonv600e)] <- "nonV600E_V600E"
```

```{r}
MTAB_counts_wt_vs_v600e_nonv600e <- MTAB_counts
rownames(MTAB_counts_wt_vs_v600e_nonv600e) <- MTAB_counts_wt_vs_v600e_nonv600e$ENSG.ID
MTAB_counts_wt_vs_v600e_nonv600e <- MTAB_counts_wt_vs_v600e_nonv600e[, -1]

MTAB_counts_wt_vs_v600e_nonv600e <- MTAB_counts_wt_vs_v600e_nonv600e[, colnames(MTAB_counts_wt_vs_v600e_nonv600e) %in% metadata_MTAB_wt_vs_v600e_nonv600e$`RNA Tumor Sample Barcode`]

metadata_MTAB_wt_vs_v600e_nonv600e <- metadata_MTAB_wt_vs_v600e_nonv600e[
  metadata_MTAB_wt_vs_v600e_nonv600e$`RNA Tumor Sample Barcode` %in% colnames(MTAB_counts_wt_vs_v600e_nonv600e),]
```


```{r}
colnames(metadata_MTAB_wt_vs_v600e_nonv600e)[40] <- "MSI_status"
colnames(metadata_MTAB_wt_vs_v600e_nonv600e)[33] <- "Tumour_Cell_Content_Pathology"
colnames(metadata_MTAB_wt_vs_v600e_nonv600e)[17] <- "Age_at_diagnosis"
colnames(metadata_MTAB_wt_vs_v600e_nonv600e)[21] <- "Primary_Site_Disease"
colnames(metadata_MTAB_wt_vs_v600e_nonv600e)[23] <- "Tumour_Site"
```




```{r}
dge_wt_vs_v600e_nonv600e <- DGEList(counts = MTAB_counts_wt_vs_v600e_nonv600e)
```

```{r}
dge_wt_vs_v600e_nonv600e <- calcNormFactors(dge_wt_vs_v600e_nonv600e)
dim(dge_wt_vs_v600e_nonv600e)
```

```{r}
cutoff <- 18
drop_wt_vs_v600e_nonv600e <- which(apply(cpm(dge_wt_vs_v600e_nonv600e), 1, max) < cutoff)
dge_drop_wt_vs_v600e_nonv600e <- dge_wt_vs_v600e_nonv600e[-drop_wt_vs_v600e_nonv600e,] 
dim(dge_drop_wt_vs_v600e_nonv600e)
```


```{r}
design_wt_vs_v600e_nonv600e <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation, data = metadata_MTAB_wt_vs_v600e_nonv600e)

```

```{r}
v_wt_vs_v600e_nonv600e <- voom(dge_drop_wt_vs_v600e_nonv600e, design_wt_vs_v600e_nonv600e, plot = TRUE)
```


```{r}
fit_wt_vs_v600e_nonv600e <- lmFit(v_wt_vs_v600e_nonv600e, design_wt_vs_v600e_nonv600e)
fit_wt_vs_v600e_nonv600e <- eBayes(fit_wt_vs_v600e_nonv600e)
```

```{r}
topTable_wt_vs_v600e_nonv600e <- topTable(fit_wt_vs_v600e_nonv600e,
                           coef = "mutationwt",
                           adjust.method = "BH",
                           number = Inf)
topTable_wt_vs_v600e_nonv600e$Gene <- rownames(topTable_wt_vs_v600e_nonv600e)
```

```{r}
topTable_wt_vs_v600e_nonv600e$Gene <- mapIds(org.Hs.eg.db,
                                             keys = rownames(topTable_wt_vs_v600e_nonv600e),
                                             column = "SYMBOL",
                                             keytype = "ENTREZID",
                                             multiVals = "first")


# topTable_wt_vs_v600e_nonv600e <- topTable_wt_vs_v600e_nonv600e[!is.na(topTable_wt_vs_v600e_nonv600e$Gene), ]
# topTable_nonv600e_vs_wt_v600e <- topTable_nonv600e_vs_wt_v600e[!duplicated(topTable_nonv600e_vs_wt_v600e$Gene), ]
```


```{r}
EnhancedVolcano(topTable_wt_vs_v600e_nonv600e,
                lab = topTable_wt_vs_v600e_nonv600e$Gene,
                x = "logFC",
                y = "P.Value",
                pCutoff = 0.05,
                FCcutoff = 1,
                title = "Limma-Voom: wt vs V600E/nonV600E",
                pointSize = 2.0,
                labSize = 3.0)
```
















#------------------------------------------------------------------------------------------------------------------------------------













#REGULON ANALYSIS
```{r}
network_MTAB <- read.delim("/CCBdata/projects/BRAF_regulon/Aracne_COAD_output/network.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
network_MTAB <- network_MTAB[, 1:3]
write.table(network_MTAB, "network.adj", sep = "\t",
            row.names = FALSE, col.names = FALSE, quote = FALSE)
Expression_Data_MTAB <- read_tsv("MTAB_ARACNE_samples.tsv")
Expression_Data_MTAB <- as.data.frame(Expression_Data_MTAB)
rownames(Expression_Data_MTAB) <- Expression_Data_MTAB[[1]]
Expression_Data_MTAB[[1]] <- NULL
Expression_Data_MTAB <- as.matrix(Expression_Data_MTAB)
```

```{r}
VIPER_MTAB_metadata <- metadata_MTAB_mutation %>% filter(!(metadata_MTAB_mutation$`Anatomic Organ Subdivision` == "Rectum"))
vector_mutation <- c("p.Val640Glu", "wt")
VIPER_MTAB_metadata$mutation[!(VIPER_MTAB_metadata$mutation %in% vector_mutation)] <- "nonV600E"
VIPER_MTAB_metadata$mutation[VIPER_MTAB_metadata$mutation == "p.Val640Glu"] <- "V600E"
```

```{r}

sample_ids_VIPER <- VIPER_MTAB_metadata$`RNA Tumor Sample Barcode`
Expression_Data_MTAB <- Expression_Data_MTAB[, colnames(Expression_Data_MTAB) %in% sample_ids_VIPER]
Expression_Data_MTAB <- Expression_Data_MTAB[, match(sample_ids_VIPER, colnames(Expression_Data_MTAB))]

```

```{r}
phenotype_VIPER <- VIPER_MTAB_metadata$mutation 
names(phenotype_VIPER) <- VIPER_MTAB_metadata$`RNA Tumor Sample Barcode`
```
##V600 VS NONV600E
```{r}
keep <- phenotype_VIPER %in% c("V600E", "nonV600E")
phenotype_VIPER <- phenotype_VIPER[keep]
Expression_Data_MTAB_V600E_nonV600E <- Expression_Data_MTAB[, names(phenotype_VIPER)]
```

```{r}
library(Biobase)
pheno_df <- data.frame(group = phenotype_VIPER)
rownames(pheno_df) <- names(phenotype_VIPER)
eset_V600E_nonV600E <- ExpressionSet(assayData = Expression_Data_MTAB_V600E_nonV600E, phenoData = AnnotatedDataFrame(pheno_df))

```

```{r}
regulon_object <- aracne2regulon("network.adj", Expression_Data_MTAB, verbose = FALSE)
```

```{r}
signature <- rowTtest(eset_V600E_nonV600E, "group", "V600E", "nonV600E")
signature <- (qnorm(signature$p.value / 2, lower.tail = FALSE) * sign(signature$statistic))[, 1]
```


```{r}
nullmodel <- ttestNull(eset_V600E_nonV600E, "group", "V600E", "nonV600E",
                       per = 1000, repos = TRUE, verbose = TRUE)
```


```{r}
mrs <- msviper(signature, regulon_object, nullmodel, verbose = TRUE)
summary(mrs)
```

```{r}
# Convert msVIPER summary to a data frame
mrs_df <- summary(mrs, 50)

# Add rownames as a column for TF labels
mrs_df$TF <- rownames(mrs_df)

# Sort by NES (optional: by abs(NES) or FDR)
mrs_df <- mrs_df[order(mrs_df$NES), ]  # from most negative to most positive

# Set factor levels for ordering in the plot
mrs_df$TF <- factor(mrs_df$TF, levels = mrs_df$TF)

```

```{r}
ggplot(mrs_df, aes(x = TF, y = NES, fill = NES)) +
  geom_col() +
  coord_flip() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0,
                       name = "NES") +
  theme_minimal(base_size = 12) +
  labs(title = "Top Master Regulators (VIPER) V600E vs nonV600E",
       x = "Regulator",
       y = "Normalized Enrichment Score (NES)") +
  theme(axis.text.y = element_text(size = 10),
        legend.position = "right")
```


```{r}
mrs <- ledge(mrs)
mrs_summary <- summary(mrs)
```






```{r}
phenotype_VIPER <- VIPER_MTAB_metadata$mutation 
names(phenotype_VIPER) <- VIPER_MTAB_metadata$`RNA Tumor Sample Barcode`
```

```{r}
keep <- phenotype_VIPER %in% c("V600E", "wt")
phenotype_VIPER <- phenotype_VIPER[keep]
Expression_Data_MTAB_V600E_wt <- Expression_Data_MTAB[, names(phenotype_VIPER)]
```

```{r}
library(Biobase)
pheno_df <- data.frame(group = phenotype_VIPER)
rownames(pheno_df) <- names(phenotype_VIPER)
eset_V600E_wt <- ExpressionSet(assayData = Expression_Data_MTAB_V600E_wt, phenoData = AnnotatedDataFrame(pheno_df))

```

```{r}
# regulon_object <- aracne2regulon("network.adj", Expression_Data_MTAB, verbose = FALSE)
```
##V600 VS wt
```{r}
signature_V600E_wt <- rowTtest(eset_V600E_wt, "group", "V600E", "wt")
signature_V600E_wt <- (qnorm(signature_V600E_wt$p.value / 2, lower.tail = FALSE) * sign(signature_V600E_wt$statistic))[, 1]
```


```{r}
nullmodel_V600E_WT <- ttestNull(eset_V600E_wt, "group", "V600E", "wt",
                       per = 1000, repos = TRUE, verbose = TRUE)
```


```{r}
mrs_V600E_wt <- msviper(signature_V600E_wt, regulon_object, nullmodel_V600E_WT, verbose = TRUE)
summary(mrs_V600E_wt)
```

```{r}
# Convert msVIPER summary to a data frame
mrs_df_V600E_WT <- summary(mrs_V600E_wt, 50)

# Add rownames as a column for TF labels
mrs_df_V600E_WT$TF <- rownames(mrs_df_V600E_WT)

# Sort by NES (optional: by abs(NES) or FDR)
mrs_df_V600E_WT <- mrs_df_V600E_WT[order(mrs_df_V600E_WT$NES), ]  # from most negative to most positive

# Set factor levels for ordering in the plot
mrs_df_V600E_WT$TF <- factor(mrs_df_V600E_WT$TF, levels = mrs_df_V600E_WT$TF)

```

```{r}
ggplot(mrs_df_V600E_WT, aes(x = TF, y = NES, fill = NES)) +
  geom_col() +
  coord_flip() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0,
                       name = "NES") +
  theme_minimal(base_size = 12) +
  labs(title = "Top Master Regulators (VIPER) V600E vs WT",
       x = "Regulator",
       y = "Normalized Enrichment Score (NES)") +
  theme(axis.text.y = element_text(size = 10),
        legend.position = "right")
```

```{r}
mrs_V600E_wt <- ledge(mrs_V600E_wt)
mrs_summary <- summary(mrs_V600E_wt)
```






```{r}
phenotype_VIPER <- VIPER_MTAB_metadata$mutation 
names(phenotype_VIPER) <- VIPER_MTAB_metadata$`RNA Tumor Sample Barcode`
```

```{r}
keep <- phenotype_VIPER %in% c("nonV600E", "wt")
phenotype_VIPER <- phenotype_VIPER[keep]
Expression_Data_MTAB_nonV600E_wt <- Expression_Data_MTAB[, names(phenotype_VIPER)]
```

```{r}
library(Biobase)
pheno_df <- data.frame(group = phenotype_VIPER)
rownames(pheno_df) <- names(phenotype_VIPER)
eset_nonV600E_wt <- ExpressionSet(assayData = Expression_Data_MTAB_nonV600E_wt, phenoData = AnnotatedDataFrame(pheno_df))

```

```{r}
# regulon_object <- aracne2regulon("network.adj", Expression_Data_MTAB, verbose = FALSE)
```
##nonV600 VS wt
```{r}
signature_nonV600E_wt <- rowTtest(eset_nonV600E_wt, "group", "nonV600E", "wt")
signature_nonV600E_wt <- (qnorm(signature_nonV600E_wt$p.value / 2, lower.tail = FALSE) * sign(signature_nonV600E_wt$statistic))[, 1]
```


```{r}
nullmodel_nonV600E_WT <- ttestNull(eset_nonV600E_wt, "group", "nonV600E", "wt",
                       per = 1000, repos = TRUE, verbose = TRUE)
```

```{r}
mrs_nonV600E_wt <- msviper(signature_nonV600E_wt, regulon_object, nullmodel_nonV600E_WT, verbose = TRUE)
summary(mrs_nonV600E_wt)
```
```{r}
# Convert msVIPER summary to a data frame
mrs_df_nonV600E_WT <- summary(mrs_nonV600E_wt, 50)

# Add rownames as a column for TF labels
mrs_df_nonV600E_WT$TF <- rownames(mrs_df_nonV600E_WT)

# Sort by NES (optional: by abs(NES) or FDR)
mrs_df_nonV600E_WT <- mrs_df_nonV600E_WT[order(mrs_df_nonV600E_WT$NES), ]  # from most negative to most positive

# Set factor levels for ordering in the plot
mrs_df_nonV600E_WT$TF <- factor(mrs_df_nonV600E_WT$TF, levels = mrs_df_nonV600E_WT$TF)

```

```{r}
ggplot(mrs_df_nonV600E_WT, aes(x = TF, y = NES, fill = NES)) +
  geom_col() +
  coord_flip() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0,
                       name = "NES") +
  theme_minimal(base_size = 12) +
  labs(title = "Top Master Regulators (VIPER) nonV600E vs WT",
       x = "Regulator",
       y = "Normalized Enrichment Score (NES)") +
  theme(axis.text.y = element_text(size = 10),
        legend.position = "right")
```


```{r}
mrs_df_nonV600E_WT <- ledge(mrs_nonV600E_wt)
mrs_summary_nonv600e_wt <- summary(mrs_df_nonV600E_WT)
```


























```{r}
regulon_object <- aracne2regulon("/CCBdata/projects/BRAF_regulon/Aracne_COAD_output/network.txt", Expression_Data_MTAB)
```


```{r}
viper_MTAB <- viper(Expression_Data_MTAB, regulon_object, method = "scale")
```


```{r}
metadata_MTAB_no_Rectum <- metadata_MTAB_mutation %>% filter(!metadata_MTAB_mutation$`Anatomic Organ Subdivision` == "Rectum")
# metadata_MTAB_no_Rectum <- metadata_MTAB_no_Rectum %>% filter(!metadata_MTAB_no_Rectum$mutation == "wt")
```

```{r}
vector_mut_regulon <- c("p.Val640Glu", "wt")
metadata_MTAB_no_Rectum$mutation[!(metadata_MTAB_no_Rectum$mutation %in% vector_mut_regulon)] <- "nonV600E"
```



```{r}
annotation_col <- data.frame(BRAF = metadata_MTAB_no_Rectum$mutation)
rownames(annotation_col) <- metadata_MTAB_no_Rectum$`RNA Tumor Sample Barcode`
```


```{r}
library(pheatmap)

# Optional: choose top variable proteins
top_var <- apply(viper_MTAB, 1, var)
top_TFs <- names(sort(top_var, decreasing = TRUE))[1:50]

pheatmap(viper_MTAB[top_TFs, ],
         annotation_col = annotation_col,
         show_colnames = FALSE,cluster_cols = TRUE, cluster_rows = FALSE,
         clustering_method = "ward.D",
         main = "VIPER Inferred TF Activity (Top 50)")
```



```{r}
library(viper)

# Grouping
group_labels <- annotation_col$BRAF
names(group_labels) <- rownames(annotation_col)

# Comparison: V600E vs WT
sel_V600E <- names(group_labels[group_labels == "p.Val640Glu"])
sel_WT <- names(group_labels[group_labels == "nonV600E"])

# Signature: logFC between V600E and WT (based on VIPER scores)
viper_diff <- rowMeans(viper_MTAB[, sel_V600E]) - rowMeans(viper_MTAB[, sel_WT])
signature <- (viper_diff)
names(signature) <- rownames(viper_MTAB)

# Run msviper
mrs <- msviper(signature, regulon_object, verbose = TRUE)
```


```{r}
library(ggplot2)

res <- summary(mrs_V600E_vs_WT.nonV600E)
res$TF <- rownames(res)

ggplot(res, aes(x = NES, y = -log10(p.value))) +
  geom_point(aes(color = p.value < 0.05)) +
  geom_text(aes(label = ifelse(p.value < 0.01, TF, "")), hjust = 1.1, vjust = 1.1, size = 2.5) +
  theme_minimal() +
  labs(title = "Master Regulators: V600E vs nonv600E", x = "NES (Normalized Enrichment Score)", y = "-log10(p-value)")
```


```{r}
top_mrs <- head(rownames(res[order(res$p.value), ]), 20)
pheatmap(viper_MTAB[top_mrs, ], annotation_col = annotation_col,
         show_colnames = FALSE, main = "Top Master Regulators by VIPER")

```



```{r}
common_samples <- intersect(metadata_MTAB_no_Rectum$`RNA Tumor Sample Barcode`, colnames(viper_MTAB))

# Subset both data sets
metadata_aligned <- metadata_MTAB_no_Rectum[metadata_MTAB_no_Rectum$`RNA Tumor Sample Barcode` %in% common_samples, ]
viper_aligned <- viper_MTAB[, common_samples]

# Reorder to make sure row order in metadata matches column order in viper
metadata_aligned <- metadata_aligned[match(colnames(viper_aligned), metadata_aligned$`RNA Tumor Sample Barcode`), ]

```

```{r}
boxplot(viper_aligned["IDH1", ] ~ metadata_aligned$mutation,
        main = "TF Activity of  TFAM by BRAF Mutation",
        ylab = "VIPER Inferred Activity (scaled)",
        xlab = "BRAF Mutation",
        col = c("skyblue", "tomato", "gray"))
```

##arnau meeting

perform limma voom. paper arnau sent (edger)
get the diff exp genes, sort the genes.
from there we use msviper.
enrichment regulons.





#MSVIPER
##v600e vs wt_nonv600e
```{r}
# Use t-statistics as expression signature
gene_signature <- topTable_v600e$t
names(gene_signature) <- topTable_v600e$Gene
gene_signature <- sort(gene_signature, decreasing = TRUE)
```

```{r}
msviper_v600e <- msviper(gene_signature, regulon_object, verbose = FALSE)
```

```{r}
p1 <- sort(msviper_v600e$es$p.value)
MRs <- names(p1)[p1 < 0.00001]
plot(msviper_v600e, 50)
```
Each row represents one putative regulator (gene symbol in the "Set" column), inferred by VIPER/msVIPER.
Each vertical line represents a target gene of that MR.
The color indicates expression/activity:
Red = activated target (positive enrichment)
Blue = repressed target (negative enrichment)
The pattern shows how the targets of this MR behave in your experiment (e.g., comparing mutant vs. wildtype).
Middle columns
p-value: Statistical significance of enrichment for that MR.
Set: The name of the MR.
Right heatmap ("Act"/"Exp")
A visual summary of activity:
"Act" column shows the inferred activity of the MR (red = activated, blue = repressed).
"Exp" column shows the actual expression of the MR gene itself.




##nonv600e vs wt_v600e
```{r}
# Use t-statistics as expression signature
gene_signature_nonv600e <- topTable_nonv600e_vs_wt_v600e$t
names(gene_signature_nonv600e) <- topTable_nonv600e_vs_wt_v600e$Gene
gene_signature_nonv600e <- sort(gene_signature_nonv600e, decreasing = TRUE)
```

```{r}
msviper_nonv600e <- msviper(gene_signature_nonv600e, regulon_object, verbose = FALSE)
```

```{r}
p2 <- sort(msviper_nonv600e$es$p.value)
# MRs_nonv600e <- names(p1)[p1 < 0.00001]
plot(msviper_nonv600e, 20)
```

##wt vs nonv600e_v600e
```{r}
# Use t-statistics as expression signature
gene_signature_wt <- topTable_wt_vs_v600e_nonv600e$t
names(gene_signature_wt) <- topTable_wt_vs_v600e_nonv600e$Gene
gene_signature_wt <- sort(gene_signature_wt, decreasing = TRUE)
```

```{r}
msviper_wt <- msviper(gene_signature_wt, regulon_object, verbose = FALSE)
```

```{r}
p3 <- sort(msviper_wt$es$p.value)
# MRs_nonv600e <- names(p1)[p1 < 0.00001]
plot(msviper_wt, 50)
```






#LIMMAxVIPER

##V600E VS WT/nonV600E
```{r}
dge_V600E_vs_WT.nonV600E <- DGEList(counts = MTAB_counts_v600e_vs_wt_nonv600e)

dge_V600E_vs_WT.nonV600E <- calcNormFactors(dge_V600E_vs_WT.nonV600E)

cutoff <- 18
keep_genes_V600E_vs_WT.nonV600E <- apply(cpm(dge_V600E_vs_WT.nonV600E), 1, max) >= cutoff
dge_V600E_vs_WT.nonV600E_filtered <- dge[keep_genes_V600E_vs_WT.nonV600E, ]
```

```{r}
metadata_MTAB_v600e_vs_wt_nonv600e$mutation <- relevel(factor(metadata_MTAB_v600e_vs_wt_nonv600e$mutation), ref = "nonV600E_wt")

design_V600E_vs_WT.nonV600E <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation, data = metadata_MTAB_v600e_vs_wt_nonv600e)
```

```{r}
v_V600E_vs_WT.nonV600E <- voom(dge_V600E_vs_WT.nonV600E_filtered, design_V600E_vs_WT.nonV600E, plot = TRUE)
fit_V600E_vs_WT.nonV600E <- lmFit(v_V600E_vs_WT.nonV600E, design_V600E_vs_WT.nonV600E)
fit_V600E_vs_WT.nonV600E <- eBayes(fit_V600E_vs_WT.nonV600E)

topTable_V600E_vs_WT.nonV600E <- topTable(fit_V600E_vs_WT.nonV600E, coef = "mutationp.Val640Glu", adjust.method = "BH", number = Inf)
topTable_V600E_vs_WT.nonV600E$ENSEMBL <- rownames(topTable_V600E_vs_WT.nonV600E)

```

```{r}
# Load GENCODE v36 annotations
gencode_v36 <- rtracklayer::import("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.annotation.gtf")
entrez_annot_v36 <- data.table::fread("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.metadata.EntrezGene.gz",
                          col.names = c("transcript_id", "entrez_id"))

# Build mapping
temp_v36 <- gencode_v36 %>%
  as.data.frame() %>%
  filter(type == "transcript") %>%
  dplyr::select(gene_id, transcript_id) %>%
  distinct(transcript_id, .keep_all = TRUE) %>%
  merge(entrez_annot_v36, by = "transcript_id", all.x = TRUE)

gencode_v36_genes <- gencode_v36[gencode_v36$type == "gene"]
gencode_v36_genes$entrez_id <- temp_v36$entrez_id[match(gencode_v36_genes$gene_id, temp_v36$gene_id, nomatch = NA)]

# Merge with DE table to get gene names
annot_df <- as.data.frame(gencode_v36_genes)[, c("gene_id", "entrez_id", "gene_name")]
annot_df$gene_id <- sub("\\..*", "", annot_df$gene_id)
topTable_V600E_vs_WT.nonV600E$gene_name <- annot_df$gene_name[match(topTable_V600E_vs_WT.nonV600E$ENSEMBL, annot_df$gene_id)]

# Filter valid genes
topTable_V600E_vs_WT.nonV600E <- topTable_V600E_vs_WT.nonV600E[!is.na(topTable_V600E_vs_WT.nonV600E$gene_name), ]
topTable_V600E_vs_WT.nonV600E <- topTable_V600E_vs_WT.nonV600E[!duplicated(topTable_V600E_vs_WT.nonV600E$gene_name), ]
```


```{r}
topTable_V600E_vs_WT.nonV600E$zscore <- qnorm(topTable_V600E_vs_WT.nonV600E$P.Value / 2, lower.tail = FALSE) * sign(topTable_V600E_vs_WT.nonV600E$t)
signature_V600E_vs_WT.nonV600E <- setNames(topTable_V600E_vs_WT.nonV600E$zscore, topTable_V600E_vs_WT.nonV600E$gene_name)
```


```{r}
# set.seed(123)
# n_perm <- 1000
# genes <- rownames(v_V600E_vs_WT.nonV600E$E)
# null_matrix_V600E_vs_WT.nonV600E <- matrix(NA, nrow = length(genes), ncol = n_perm)
# rownames(null_matrix_V600E_vs_WT.nonV600E) <- genes
# 
# for (i in 1:n_perm) {
#   metadata_MTAB_v600e_vs_wt_nonv600e$mutation_perm <- sample(metadata_MTAB_v600e_vs_wt_nonv600e$mutation)
#   design_perm <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation_perm, 
#                               data = metadata_MTAB_v600e_vs_wt_nonv600e)
#   
#   fit_perm <- lmFit(v_V600E_vs_WT.nonV600E, design_perm)
#   fit_perm <- eBayes(fit_perm)
#   
#   null_matrix_V600E_vs_WT.nonV600E[, i] <- fit_perm$t[, "mutation_permp.Val640Glu"]
# }

# save(v_V600E_vs_WT.nonV600E, metadata_MTAB_v600e_vs_wt_nonv600e, file = "VIPER_V600E_vs_WT.nonV600E.RData")

load("Null_Matrix_V600E_vs_WT.nonV600E.RData")

# Estimate residual degrees of freedom from one model (use the smallest as conservative)

```

```{r}
# 1. Load your GENCODE v36 mapping
gencode_v36 <- rtracklayer::import("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.annotation.gtf")

gencode_v36_genes <- gencode_v36[gencode_v36$type == "gene"]
gencode_df <- as.data.frame(gencode_v36_genes)[, c("gene_id", "gene_name")]
gencode_df <- gencode_df[!duplicated(gencode_df$gene_id), ]

# 2. Map ENSEMBL IDs to gene names in null_matrix
ensembl_ids <- rownames(null_matrix_V600E_vs_WT.nonV600E)
gencode_df$gene_id <- sub("\\..*", "", gencode_df$gene_id)
gene_names <- gencode_df$gene_name[match(ensembl_ids, gencode_df$gene_id)]

# 3. Filter and rename null_matrix
null_matrix_mapped <- null_matrix_V600E_vs_WT.nonV600E[!is.na(gene_names), ]
rownames(null_matrix_mapped) <- gene_names[!is.na(gene_names)]

# 4. Remove duplicates (if any)
null_matrix_mapped_V600E_vs_WT.nonV600E <- null_matrix_mapped[!duplicated(rownames(null_matrix_mapped)), ]

df_residual <- min(fit_perm$df.residual)

# Convert entire matrix of t-stats to two-sided p-values
p_null <- 2 * pt(-abs(null_matrix_mapped_V600E_vs_WT.nonV600E), df = df_residual)

# Convert to z-scores (same formula as for signature)
null_matrix_mapped_V600E_vs_WT.nonV600E_ZSCORE <- qnorm(p_null / 2, lower.tail = FALSE) * sign(null_matrix_mapped_V600E_vs_WT.nonV600E)
```

```{r}
network_MTAB <- read.delim("/CCBdata/projects/BRAF_regulon/Aracne_COAD_output_MTAB_protein_coding/network.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
network_MTAB <- network_MTAB[, 1:3]
write.table(network_MTAB, "network_MTAB.adj", sep = "\t",
            row.names = FALSE, col.names = FALSE, quote = FALSE)
Expression_Data_MTAB <- read_tsv("MTAB_ARACNE_samples.tsv")
Expression_Data_MTAB <- as.data.frame(Expression_Data_MTAB)
rownames(Expression_Data_MTAB) <- Expression_Data_MTAB[[1]]
Expression_Data_MTAB[[1]] <- NULL
Expression_Data_MTAB <- as.matrix(Expression_Data_MTAB) 
```

```{r}
# Make sure you previously created regulon_object from ARACNe output
regulon_object <- aracne2regulon("network_MTAB.adj", Expression_Data_MTAB, verbose = FALSE)

# Run msVIPER
mrs_V600E_vs_WT.nonV600E <- msviper(signature_V600E_vs_WT.nonV600E, regulon_object, nullmodel = null_matrix_mapped_V600E_vs_WT.nonV600E_ZSCORE, verbose = TRUE)

```

```{r}
# Top results
mrs_V600E_vs_WT.nonV600E_df <- summary(mrs_V600E_vs_WT.nonV600E, 50)
mrs_V600E_vs_WT.nonV600E_df$TF <- rownames(mrs_V600E_vs_WT.nonV600E_df)
mrs_V600E_vs_WT.nonV600E_df <- mrs_V600E_vs_WT.nonV600E_df[order(mrs_V600E_vs_WT.nonV600E_df$NES), ]
mrs_V600E_vs_WT.nonV600E_df$TF <- factor(mrs_V600E_vs_WT.nonV600E_df$TF, levels = mrs_V600E_vs_WT.nonV600E_df$TF)

mrs_V600E_vs_WT.nonV600E_df <- mrs_V600E_vs_WT.nonV600E_df %>%
  mutate(logFDR = -log10(FDR),
         TF = factor(TF, levels = TF[order(NES)]),
         p_value = p.value)
# Plot
ggplot(mrs_V600E_vs_WT.nonV600E_df, aes(x = NES, y = TF)) +
  geom_point(aes(size = logFDR, color = NES)) +
  scale_color_gradient2(low = "skyblue", mid = "white", high = "salmon", midpoint = 0,
                        name = "NES") +
  scale_size_continuous(name = "-log10(FDR)", range = c(2, 8)) +
  geom_vline(xintercept = 0, linetype = "dashed", color = "gray60") +
  theme_minimal(base_size = 13) +
  labs(title = "VIPER: BRAFV600E vs WT/nonV600E",
       x = "Normalized Enrichment Score (NES)",
       y = "Transcription Factor") +
  theme(axis.text.y = element_text(size = 11),
        plot.title = element_text(face = "bold", hjust = 0.5),
        legend.position = "right")

```

Shadow analysis
```{r}
mrshadow_V600E_vs_WT.nonV600E <- shadow(mrs_V600E_vs_WT.nonV600E, regulators = 25, verbose = FALSE)
summary(mrshadow_V600E_vs_WT.nonV600E)
```
###TCGA regulon
```{r}
network_TCGA <- read.delim("/CCBdata/projects/BRAF_regulon/Aracne_COAD_output_TCGA_protein_coding/network.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
network_TCGA <- network_TCGA[, 1:3]
write.table(network_TCGA, "network_TCGA.adj", sep = "\t",
            row.names = FALSE, col.names = FALSE, quote = FALSE)
Expression_Data_TCGA <- read_tsv("ARACNe_log2transformed_TPM_tcga_473samples.tsv")
Expression_Data_TCGA <- as.data.frame(Expression_Data_TCGA)
Expression_Data_TCGA <- Expression_Data_TCGA[!duplicated(Expression_Data_TCGA[[1]]), ]
rownames(Expression_Data_TCGA) <- Expression_Data_TCGA[[1]]
Expression_Data_TCGA[[1]] <- NULL
Expression_Data_TCGA <- as.matrix(Expression_Data_TCGA) 
```


```{r}
regulon_object_TCGA <- aracne2regulon("network_TCGA.adj", Expression_Data_TCGA, verbose = FALSE)

# Run msVIPER
mrs_V600E_vs_WT.nonV600E_TCGA <- msviper(signature_V600E_vs_WT.nonV600E, regulon_object_TCGA, nullmodel = null_matrix_mapped_V600E_vs_WT.nonV600E_ZSCORE, verbose = TRUE)

```

```{r}
# Top results
mrs_V600E_vs_WT.nonV600E_TCGA_df <- summary(mrs_V600E_vs_WT.nonV600E_TCGA, 50)
mrs_V600E_vs_WT.nonV600E_TCGA_df$TF <- rownames(mrs_V600E_vs_WT.nonV600E_TCGA_df)
mrs_V600E_vs_WT.nonV600E_TCGA_df <- mrs_V600E_vs_WT.nonV600E_TCGA_df[order(mrs_V600E_vs_WT.nonV600E_TCGA_df$NES), ]
mrs_V600E_vs_WT.nonV600E_TCGA_df$TF <- factor(mrs_V600E_vs_WT.nonV600E_TCGA_df$TF, levels = mrs_V600E_vs_WT.nonV600E_TCGA_df$TF)

mrs_V600E_vs_WT.nonV600E_TCGA_df <- mrs_V600E_vs_WT.nonV600E_TCGA_df %>%
  mutate(logFDR = -log10(FDR),
         TF = factor(TF, levels = TF[order(NES)]))
# Plot
ggplot(mrs_V600E_vs_WT.nonV600E_TCGA_df, aes(x = NES, y = TF)) +
  geom_point(aes(size = logFDR, color = NES)) +
  scale_color_gradient2(low = "skyblue", mid = "white", high = "salmon", midpoint = 0,
                        name = "NES") +
  scale_size_continuous(name = "-log10(FDR)", range = c(2, 8)) +
  geom_vline(xintercept = 0, linetype = "dashed", color = "gray60") +
  theme_minimal(base_size = 13) +
  labs(title = "VIPER: BRAFV600E vs WT/nonV600E TCGA",
       x = "Normalized Enrichment Score (NES)",
       y = "Transcription Factor") +
  theme(axis.text.y = element_text(size = 11),
        plot.title = element_text(face = "bold", hjust = 0.5),
        legend.position = "right")

```























##nonV600E VS WT/V600E
```{r}
dge_nonV600E_vs_WT.V600E <- DGEList(counts = MTAB_counts_nonv600e_vs_wt_v600e)

dge_nonV600E_vs_WT.V600E <- calcNormFactors(dge_nonV600E_vs_WT.V600E)

cutoff <- 18
keep_genes_nonV600E_vs_WT.V600E <- apply(cpm(dge_nonV600E_vs_WT.V600E), 1, max) >= cutoff
dge_nonV600E_vs_WT.V600E_filtered <- dge_nonV600E_vs_WT.V600E[keep_genes_nonV600E_vs_WT.V600E, ]
```

```{r}
metadata_MTAB_nonv600e_vs_wt_v600e$mutation <- relevel(factor(metadata_MTAB_nonv600e_vs_wt_v600e$mutation), ref = "V600E_wt")

design_nonV600E_vs_WT.V600E <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation, data = metadata_MTAB_nonv600e_vs_wt_v600e)
```

```{r}
v_nonV600E_vs_WT.V600E <- voom(dge_nonV600E_vs_WT.V600E_filtered, design_nonV600E_vs_WT.V600E, plot = TRUE)
fit_nonV600E_vs_WT.V600E <- lmFit(v_nonV600E_vs_WT.V600E, design_nonV600E_vs_WT.V600E)
fit_nonV600E_vs_WT.V600E <- eBayes(fit_nonV600E_vs_WT.V600E)

topTable_nonV600E_vs_WT.V600E <- topTable(fit_nonV600E_vs_WT.V600E, coef = "mutationnonV600E", adjust.method = "BH", number = Inf)
topTable_nonV600E_vs_WT.V600E$ENSEMBL <- rownames(topTable_nonV600E_vs_WT.V600E)
```

```{r}
# Load GENCODE v36 annotations
gencode_v36 <- rtracklayer::import("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.annotation.gtf")
entrez_annot_v36 <- data.table::fread("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.metadata.EntrezGene.gz",
                          col.names = c("transcript_id", "entrez_id"))

# Build mapping
temp_v36 <- gencode_v36 %>%
  as.data.frame() %>%
  filter(type == "transcript") %>%
  dplyr::select(gene_id, transcript_id) %>%
  distinct(transcript_id, .keep_all = TRUE) %>%
  merge(entrez_annot_v36, by = "transcript_id", all.x = TRUE)

gencode_v36_genes <- gencode_v36[gencode_v36$type == "gene"]
gencode_v36_genes$entrez_id <- temp_v36$entrez_id[match(gencode_v36_genes$gene_id, temp_v36$gene_id, nomatch = NA)]

# Merge with DE table to get gene names
annot_df <- as.data.frame(gencode_v36_genes)[, c("gene_id", "entrez_id", "gene_name")]
annot_df$gene_id <- sub("\\..*", "", annot_df$gene_id)
topTable_nonV600E_vs_WT.V600E$gene_name <- annot_df$gene_name[match(topTable_nonV600E_vs_WT.V600E$ENSEMBL, annot_df$entrez_id)]

# Filter valid genes
topTable_nonV600E_vs_WT.V600E <- topTable_nonV600E_vs_WT.V600E[!is.na(topTable_nonV600E_vs_WT.V600E$gene_name), ]
topTable_nonV600E_vs_WT.V600E <- topTable_nonV600E_vs_WT.V600E[!duplicated(topTable_nonV600E_vs_WT.V600E$gene_name), ]
```


```{r}
topTable_nonV600E_vs_WT.V600E$zscore <- qnorm(topTable_nonV600E_vs_WT.V600E$P.Value / 2, lower.tail = FALSE) * sign(topTable_nonV600E_vs_WT.V600E$t)
signature_nonV600E_vs_WT.V600E <- setNames(topTable_nonV600E_vs_WT.V600E$zscore, topTable_nonV600E_vs_WT.V600E$gene_name)
```

```{r}
# set.seed(123)
# n_perm <- 1000
# genes <- rownames(v_nonV600E_vs_WT.V600E$E)
# null_matrix_nonV600E_vs_WT.V600E <- matrix(NA, nrow = length(genes), ncol = n_perm)
# rownames(null_matrix_nonV600E_vs_WT.V600E) <- genes
# 
# for (i in 1:n_perm) {
#   metadata_MTAB_nonv600e_vs_wt_v600e$mutation_perm <- sample(metadata_MTAB_nonv600e_vs_wt_v600e$mutation)
#   design_perm <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation_perm,
#                               data = metadata_MTAB_nonv600e_vs_wt_v600e)
# 
#   fit_perm <- lmFit(v_nonV600E_vs_WT.V600E, design_perm)
#   fit_perm <- eBayes(fit_perm)
# 
#   null_matrix_nonV600E_vs_WT.V600E[, i] <- fit_perm$t[, "mutation_permnonV600E"]
# }
# 
# save(v_nonV600E_vs_WT.V600E, metadata_MTAB_nonv600e_vs_wt_v600e, file = "VIPER_nonV600E_vs_WT.V600E.RData")

load("Null_Matrix_nonV600E_vs_WT.V600E.RData")
```

```{r}
# 1. Load your GENCODE v36 mapping
gencode_v36 <- rtracklayer::import("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.annotation.gtf")

gencode_v36_genes <- gencode_v36[gencode_v36$type == "gene"]
gencode_df <- as.data.frame(gencode_v36_genes)[, c("gene_id", "gene_name")]
gencode_df <- gencode_df[!duplicated(gencode_df$gene_id), ]

# 2. Map ENSEMBL IDs to gene names in null_matrix
ensembl_ids <- rownames(null_matrix_nonV600E_vs_WT.V600E)
gencode_df$gene_id <- sub("\\..*", "", gencode_df$gene_id)
gene_names <- gencode_df$gene_name[match(ensembl_ids, gencode_df$gene_id)]

# 3. Filter and rename null_matrix
null_matrix_mapped <- null_matrix_nonV600E_vs_WT.V600E[!is.na(gene_names), ]
rownames(null_matrix_mapped) <- gene_names[!is.na(gene_names)]

# 4. Remove duplicates (if any)
null_matrix_mapped_nonV600E_vs_WT.V600E <- null_matrix_mapped[!duplicated(rownames(null_matrix_mapped)), ]

df_residual <- min(fit_perm$df.residual)

# Convert entire matrix of t-stats to two-sided p-values
p_null <- 2 * pt(-abs(null_matrix_mapped_nonV600E_vs_WT.V600E), df = df_residual)

# Convert to z-scores (same formula as for signature)
null_matrix_mapped_nonV600E_vs_WT.V600E_ZSCORE <- qnorm(p_null / 2, lower.tail = FALSE) * sign(null_matrix_mapped_nonV600E_vs_WT.V600E)
```

```{r}
network_MTAB <- read.delim("/CCBdata/projects/BRAF_regulon/Aracne_COAD_output_MTAB_protein_coding/network.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
network_MTAB <- network_MTAB[, 1:3]
write.table(network_MTAB, "network_MTAB.adj", sep = "\t",
            row.names = FALSE, col.names = FALSE, quote = FALSE)
Expression_Data_MTAB <- read_tsv("MTAB_ARACNE_samples.tsv")
Expression_Data_MTAB <- as.data.frame(Expression_Data_MTAB)
rownames(Expression_Data_MTAB) <- Expression_Data_MTAB[[1]]
Expression_Data_MTAB[[1]] <- NULL
Expression_Data_MTAB <- as.matrix(Expression_Data_MTAB)
```

```{r}
# Make sure you previously created regulon_object from ARACNe output
# regulon_object <- aracne2regulon("network_MTAB.adj", Expression_Data_MTAB, verbose = FALSE)

# Run msVIPER
mrs_nonV600E_vs_WT.V600E <- msviper(signature_nonV600E_vs_WT.V600E, regulon_object, nullmodel = null_matrix_nonV600E_vs_WT.V600E, verbose = TRUE)

```

```{r}
# Top results
mrs_nonV600E_vs_WT.V600E_df <- summary(mrs_nonV600E_vs_WT.V600E, 50)
mrs_nonV600E_vs_WT.V600E_df$TF <- rownames(mrs_nonV600E_vs_WT.V600E_df)
mrs_nonV600E_vs_WT.V600E_df <- mrs_nonV600E_vs_WT.V600E_df[order(mrs_nonV600E_vs_WT.V600E_df$NES), ]
mrs_nonV600E_vs_WT.V600E_df$TF <- factor(mrs_nonV600E_vs_WT.V600E_df$TF, levels = mrs_nonV600E_vs_WT.V600E_df$TF)

# Plot
ggplot(mrs_nonV600E_vs_WT.V600E_df, aes(x = TF, y = NES, fill = NES)) +
  geom_col() +
  coord_flip() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0,
                       name = "NES") +
  theme_minimal(base_size = 12) +
  labs(title = "VIPER: BRAF nonV600E vs WT/V600E",
       x = "Regulator",
       y = "Normalized Enrichment Score (NES)") +
  theme(axis.text.y = element_text(size = 10),
        legend.position = "right")

```

Shadow analysis
```{r}
mrshadow_nonV600E_vs_WT.V600E <- shadow(mrs_nonV600E_vs_WT.V600E_df, regulators = 25, verbose = FALSE)
summary(mrshadow_nonV600E_vs_WT.V600E)
```





























##WT VS V600E/nonV600E

```{r}
dge_WT_vs_V600E.nonV600E <- DGEList(counts = MTAB_counts_wt_vs_v600e_nonv600e)

dge_WT_vs_V600E.nonV600E <- calcNormFactors(dge_WT_vs_V600E.nonV600E)

cutoff <- 18
keep_genes_WT_vs_V600E.nonV600E <- apply(cpm(dge_WT_vs_V600E.nonV600E), 1, max) >= cutoff
dge_WT_vs_V600E.nonV600E_filtered <- dge_WT_vs_V600E.nonV600E[keep_genes_WT_vs_V600E.nonV600E, ]
```

```{r}
metadata_MTAB_wt_vs_v600e_nonv600e$mutation <- relevel(factor(metadata_MTAB_wt_vs_v600e_nonv600e$mutation), ref = "nonV600E_V600E")

design_WT_vs_V600E.nonV600E <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation, data = metadata_MTAB_wt_vs_v600e_nonv600e)
```

```{r}
v_WT_vs_V600E.nonV600E <- voom(dge_WT_vs_V600E.nonV600E_filtered, design_WT_vs_V600E.nonV600E, plot = TRUE)
fit_WT_vs_V600E.nonV600E <- lmFit(v_WT_vs_V600E.nonV600E, design_WT_vs_V600E.nonV600E)
fit_WT_vs_V600E.nonV600E <- eBayes(fit_WT_vs_V600E.nonV600E)

topTable_WT_vs_V600E.nonV600E <- topTable(fit_WT_vs_V600E.nonV600E, coef = "mutationwt", adjust.method = "BH", number = Inf)
topTable_WT_vs_V600E.nonV600E$ENSEMBL <- rownames(topTable_WT_vs_V600E.nonV600E)
```

```{r}
# Load GENCODE v36 annotations
gencode_v36 <- rtracklayer::import("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.annotation.gtf")
entrez_annot_v36 <- data.table::fread("/CCBdata/users/arnau/MOBER/extdata/gencode.v36.metadata.EntrezGene.gz",
                          col.names = c("transcript_id", "entrez_id"))

# Build mapping
temp_v36 <- gencode_v36 %>%
  as.data.frame() %>%
  filter(type == "transcript") %>%
  dplyr::select(gene_id, transcript_id) %>%
  distinct(transcript_id, .keep_all = TRUE) %>%
  merge(entrez_annot_v36, by = "transcript_id", all.x = TRUE)

gencode_v36_genes <- gencode_v36[gencode_v36$type == "gene"]
gencode_v36_genes$entrez_id <- temp_v36$entrez_id[match(gencode_v36_genes$gene_id, temp_v36$gene_id, nomatch = NA)]

# Merge with DE table to get gene names
annot_df <- as.data.frame(gencode_v36_genes)[, c("gene_id", "entrez_id", "gene_name")]
annot_df$gene_id <- sub("\\..*", "", annot_df$gene_id)
topTable_WT_vs_V600E.nonV600E$gene_name <- annot_df$gene_name[match(topTable_WT_vs_V600E.nonV600E$ENSEMBL, annot_df$entrez_id)]

# Filter valid genes
topTable_WT_vs_V600E.nonV600E <- topTable_WT_vs_V600E.nonV600E[!is.na(topTable_WT_vs_V600E.nonV600E$gene_name), ]
topTable_WT_vs_V600E.nonV600E <- topTable_WT_vs_V600E.nonV600E[!duplicated(topTable_WT_vs_V600E.nonV600E$gene_name), ]
```


```{r}
topTable_WT_vs_V600E.nonV600E$zscore <- qnorm(topTable_WT_vs_V600E.nonV600E$P.Value / 2, lower.tail = FALSE) * sign(topTable_WT_vs_V600E.nonV600E$t)
signature_WT_vs_V600E.nonV600E <- setNames(topTable_WT_vs_V600E.nonV600E$zscore, topTable_WT_vs_V600E.nonV600E$gene_name)
```

```{r}
# set.seed(123)
# n_perm <- 1
# genes <- rownames(v_WT_vs_V600E.nonV600E$E)
# null_matrix_WT_vs_V600E.nonV600E <- matrix(NA, nrow = length(genes), ncol = n_perm)
# rownames(null_matrix_WT_vs_V600E.nonV600E) <- genes
# 
# for (i in 1:n_perm) {
#   metadata_MTAB_wt_vs_v600e_nonv600e$mutation_perm <- sample(metadata_MTAB_wt_vs_v600e_nonv600e$mutation)
#   design_perm <- model.matrix(~ Sex + MSI_status + Tumour_Cell_Content_Pathology + Age_at_diagnosis + Tumour_Site + mutation_perm,
#                               data = metadata_MTAB_wt_vs_v600e_nonv600e)
# 
#   fit_perm <- lmFit(v_WT_vs_V600E.nonV600E, design_perm)
#   fit_perm <- eBayes(fit_perm)
# 
#   null_matrix_WT_vs_V600E.nonV600E[, i] <- fit_perm$t[, "mutation_permwt"]
# }
# 
# save(v_WT_vs_V600E.nonV600E, metadata_MTAB_wt_vs_v600e_nonv600e, file = "VIPER_WT_vs_V600E.nonV600E.RData")
```


```{r}
network_MTAB <- read.delim("/CCBdata/projects/BRAF_regulon/Aracne_COAD_output_MTAB_protein_coding/network.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)
network_MTAB <- network_MTAB[, 1:3]
write.table(network_MTAB, "network_MTAB.adj", sep = "\t",
            row.names = FALSE, col.names = FALSE, quote = FALSE)
Expression_Data_MTAB <- read_tsv("MTAB_ARACNE_samples.tsv")
Expression_Data_MTAB <- as.data.frame(Expression_Data_MTAB)
rownames(Expression_Data_MTAB) <- Expression_Data_MTAB[[1]]
Expression_Data_MTAB[[1]] <- NULL
Expression_Data_MTAB <- as.matrix(Expression_Data_MTAB)
```

```{r}
# Make sure you previously created regulon_object from ARACNe output
# regulon_object <- aracne2regulon("network_MTAB.adj", Expression_Data_MTAB, verbose = FALSE)

# Run msVIPER
mrs_nonV600E_vs_WT.V600E <- msviper(signature_nonV600E_vs_WT.V600E, regulon_object, nullmodel = null_matrix_nonV600E_vs_WT.V600E, verbose = TRUE)

```

```{r}
# Top results
mrs_nonV600E_vs_WT.V600E_df <- summary(mrs_nonV600E_vs_WT.V600E, 50)
mrs_nonV600E_vs_WT.V600E_df$TF <- rownames(mrs_nonV600E_vs_WT.V600E_df)
mrs_nonV600E_vs_WT.V600E_df <- mrs_nonV600E_vs_WT.V600E_df[order(mrs_nonV600E_vs_WT.V600E_df$NES), ]
mrs_nonV600E_vs_WT.V600E_df$TF <- factor(mrs_nonV600E_vs_WT.V600E_df$TF, levels = mrs_nonV600E_vs_WT.V600E_df$TF)

# Plot
ggplot(mrs_nonV600E_vs_WT.V600E_df, aes(x = TF, y = NES, fill = NES)) +
  geom_col() +
  coord_flip() +
  scale_fill_gradient2(low = "blue", mid = "white", high = "red", midpoint = 0,
                       name = "NES") +
  theme_minimal(base_size = 12) +
  labs(title = "VIPER: BRAF nonV600E vs WT/V600E",
       x = "Regulator",
       y = "Normalized Enrichment Score (NES)") +
  theme(axis.text.y = element_text(size = 10),
        legend.position = "right")

```

Shadow analysis
```{r}
mrshadow_nonV600E_vs_WT.V600E <- shadow(mrs_nonV600E_vs_WT.V600E_df, regulators = 25, verbose = FALSE)
summary(mrshadow_nonV600E_vs_WT.V600E)
```























